<!DOCTYPE html><html prefix="dcterms: http://purl.org/dc/terms/">
<head>
<title>Matabarcoding and DNA barcoding for Ecologists</title>
<!--Generated on Sat Sep 24 08:10:53 2016 by LaTeXML (version 0.8.0_02) http://dlmf.nist.gov/LaTeXML/.-->
<!--Document created on September 24, 2016.-->

<meta http-equiv="Content-Type" content="text/html; charset=UTF-8">
<link rel="stylesheet" href="LaTeXML.css" type="text/css">
<link rel="stylesheet" href="ltx-book.css" type="text/css">
</head>
<body>
<div class="ltx_page_main">
<div class="ltx_page_content">
<article class="ltx_document">
<h1 class="ltx_title ltx_title_document">Matabarcoding and DNA barcoding for Ecologists</h1>
<div class="ltx_authors">
<span class="ltx_creator ltx_role_author">
<span class="ltx_personname">Akifumi S. Tanabe
</span></span>
</div>
<div class="ltx_date ltx_role_creation">September 24, 2016</div>

<section id="Chx1" class="ltx_chapter">
<h2 class="ltx_title ltx_title_chapter">Preface</h2>

<div id="Chx1.p1" class="ltx_para">
<p class="ltx_p">I started the writing of this text to promote amplicon sequence assember “Assams” and utility programs package “Claident” for sequence clustering and identification.
In this text, I explain not only usage of these programs but also the methods and procedures of data collection and taxonomic identification by DNA barcoding.</p>
</div>
<div id="Chx1.p2" class="ltx_para">
<p class="ltx_p">Metabarcoding has been already widely used in bacterial reseaches, but it’s utility is not limited to bacteria.
Metabarcoding is also applicable to soil fungi, fungi inhabiting the bodies of animals and plants, aquatic planktons, and environmental DNA emitted from macro-organisms.
I explain the methods of amplification of barcoding locus from environmental DNA or metagenomes, sequencing by high-throughput sequencers (a.k.a next-generation sequencers), taxonomic assignment of nucleotide sequences (a.k.a. DNA barcoding), and observation of presence/absence of operational taxonomic units.</p>
</div>
<div id="Chx1.p3" class="ltx_para">
<p class="ltx_p">This text is distributed under Creative-Commons Attribution-ShareAlike 2.1 Japan License.
You can copy, redistribute, display this text if you designate the authorship.
You can also modify this text and distribute the modified version if you designate the authorship and apply this license or compatible license to the modified version.
You can read the license at the following URL.
<br class="ltx_break"><a href="http://creativecommons.org/licenses/by-sa/2.1/jp/" title="" class="ltx_ref">http://creativecommons.org/licenses/by-sa/2.1/jp/</a>
<br class="ltx_break">You can also ask about this license to Creative-Commons, 171 Second Street, Suite 300, San Francisco, California 94105, USA.</p>
</div>
<div id="Chx1.p4" class="ltx_para">
<p class="ltx_p">I hope that this text helps you.
I am grateful to Dr. Hirokazu Toju (Graduate Shool of Human and Environmental Sciences, Kyoto University), Dr. Satoshi Nagai (National Research Institute of Fisheries Science, Japan Fisheries Research and Education Agency), and you.</p>
</div>
</section>
<section id="Chx2" class="ltx_chapter">
<h2 class="ltx_title ltx_title_chapter">Legends</h2>

<div id="Chx2.p1" class="ltx_para">
<p class="ltx_p">In this text, the input commands to terminals and display outputs are described as below.</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
          <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># comments</span>
        </span>
<span class="ltx_p ltx_align_left">
          <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; command argument1 \</span>
        </span>
<span class="ltx_p ltx_align_left">
          <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">argument2 \</span>
        </span>
<span class="ltx_p ltx_align_left">
          <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">argument3↓</span>
        </span>
<span class="ltx_p ltx_align_left">
          <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">output of command</span>
        </span>
<span class="ltx_p ltx_align_left">
          <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; command argument1 argument2 argument3↓</span>
        </span>
<span class="ltx_p ltx_align_left">
          <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">output of command</span>
        </span>
</p>
<p class="ltx_p">In the above example, the same commands <span class="ltx_text ltx_font_typewriter">command argument1 argument2 argument3</span> were executed twice.
The outputs <span class="ltx_text ltx_font_typewriter">output of command</span> were displayed after execution.
The characters between # and line feed were comments and needless to input.
<span class="ltx_text ltx_font_typewriter">&gt;</span> and space of line head indicate the prompt of terminal.
Do not type these characters.
↓ means the end of input commands and arguments and needless to input, but you need to type Enter key to input line feed.
I use line feed within commands or arguments for viewability.
Such line feed is led by <span class="ltx_text ltx_font_typewriter">\</span>.
Therefore, the line feeds led by <span class="ltx_text ltx_font_typewriter">\</span> do not mean the end of commands or arguments, or designation to input Enter key.
Involuntary line feeds may be generated by word wrap function depending on your read environment, but do not mean the end of commands or arguments, or designation to input Enter key.</p>
</div>
<div id="Chx2.p2" class="ltx_para">
<p class="ltx_p">The file content is shown as below in this text.</p>
<p class="ltx_p ltx_align_left" style="background-color:#E6E6E6;padding-top:12pt;padding-bottom:12pt;">
<span class="ltx_p ltx_align_left">
          <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| The content of first line</span>
        </span>
<span class="ltx_p ltx_align_left">
          <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| The content of second line</span>
        </span>
</p>
<p class="ltx_p"><span class="ltx_text ltx_font_typewriter">|</span> and space of line head indicate the line head in the file, do not exist in the file and needless to input these characters.
This code is written to help you to distinguish true line feeds and involuntary line feeds.</p>
</div>
</section>
<section id="Ch0" class="ltx_chapter">
<h2 class="ltx_title ltx_title_chapter">
<span class="ltx_tag ltx_tag_chapter">Chapter 0 </span>Installing softwares and preparing analysis environment</h2>

<div id="Ch0.p1" class="ltx_para">
<p class="ltx_p">In this text, I assume Debian GNU/Linux jessie (hereafter Debian) or Ubuntu Linux 14.04 LTS (hereafter Ubuntu) as operating system.
If you use Windows PC, please install Debian or Ubuntu.
Cygwin or Windows Subsystem for Linux provided for Windows 10 build 14393 or later can be used for the following analysis, but the programs run much more slowly.
You can use CD, DVD or USB memory to boot installer of Linux.
If your PC has only one storage device, you need to reduce Windows partition by using partition resizer software such as EaseUS Partition Master or using a partition resize function contained in the installer.
You can also use newly added internal storage devices or external storage devices connected by USB.
There are several variations of Ubuntu, and I recommend Xubuntu rather than normal Ubuntu.</p>
</div>
<div id="Ch0.p2" class="ltx_para">
<p class="ltx_p">Debian and Ubuntu can be installed to Mac.
If there is no enough space, you need to resize OSX partition with the aid of Disk Utility or add storage device.
The rEFIt or rEFInd boot selecter may be required to boot Debian, Ubuntu or the installer of them on Mac.
If you install rEFIt or rEFInd to your Mac, you can boot the installer of Debian or Ubuntu from CD, DVD or USB memory.
Do not delete existing partition of OSX.
If you have enough free space, you don’t need to use Disk Utility to resize existing partition.
You can install Debian or Ubuntu to external storage devices on Mac.</p>
</div>
<div id="Ch0.p3" class="ltx_para">
<p class="ltx_p">I assume Intel64/AMD64 (x86_64) CPU machine as analysis environment.
The other CPU machine can be used for analysis, but you need to solve problems by yourself.
The 64 bits version of Debian or Ubuntu is also required because 32 bits version cannot use large memory.</p>
</div>
<section id="Ch0.S1" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">0.1 </span>Installation of Claident, Assams, databases, and the other required programs</h3>

<div id="Ch0.S1.p1" class="ltx_para">
<p class="ltx_p">Run the following commands in terminal or console as the user that can use <span class="ltx_text ltx_font_typewriter">sudo</span>.
Then, all of the required softwares will be installed.
The installer will ask password to you when <span class="ltx_text ltx_font_typewriter">sudo</span> is used.</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; mkdir -p ~/workingdirectory↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; cd ~/workingdirectory↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; wget https://www.claident.org/installClaident_Debian.sh↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; sh installClaident_Debian.sh↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; wget https://www.claident.org/installOptions_Debian.sh↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; sh installOptions_Debian.sh↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; wget https://www.claident.org/installDB_Debian.sh↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; sh installDB_Debian.sh↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; wget https://www.claident.org/installUCHIMEDB_Debian.sh↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; sh installUCHIMEDB_Debian.sh↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; cd ..↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; rm -r workingdirectory↓</span>
          </span>
</p>
<p class="ltx_p">By default, the softwares will be installed to <span class="ltx_text ltx_font_typewriter">/usr/local</span>.
In the installation, you will see <span class="ltx_text ltx_font_typewriter">Permission denied</span> error and the installer ask password to you.
If the installer continue after password input, you don’t need to care about the error.
The installer try to install without <span class="ltx_text ltx_font_typewriter">sudo</span> at first and the installation output the above error.
Then, the installer try to install using <span class="ltx_text ltx_font_typewriter">sudo</span>.</p>
</div>
<div id="Ch0.S1.p2" class="ltx_para">
<p class="ltx_p">If you need proxy to connect the internet, execute the following commands to set environment variables before execution of the installer.</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; export http_proxy=http://server.address:portnumber↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; export https_proxy=$http_proxy↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; export ftp_proxy=$http_proxy↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; export all_proxy=$http_proxy↓</span>
          </span>
</p>
<p class="ltx_p">If the proxy requires username and password, execute the following commands instead of the above commands.</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; export http_proxy=http://username:password@server.address:portnumber↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; export https_proxy=$http_proxy↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; export ftp_proxy=$http_proxy↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; export all_proxy=$http_proxy↓</span>
          </span>
</p>
</div>
<section id="Ch0.S1.SS1" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">0.1.1 </span>Upgrading to new version</h4>

<div id="Ch0.S1.SS1.p1" class="ltx_para">
<p class="ltx_p">If you want to upgrade all of the softwares and the databases, run the same commands as initial installation.
By this procedure, Assams, Claident, PEAR, VSEARCH, Metaxa and ITSx will be installed to <span class="ltx_text ltx_font_typewriter">/usr/local</span>, and NCBI BLAST+, BLAST databases for molecular identification, taxonomy databases and the other required programs will be installed to <span class="ltx_text ltx_font_typewriter">/usr/local/share/claident</span>.
NCBI BLAST+ and BLAST databases used by Claident can co-exist system wide installation of NCBI BLAST+ and BLAST databases.</p>
</div>
<div id="Ch0.S1.SS1.p2" class="ltx_para">
<p class="ltx_p">You can disable a part of upgrade like below.</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; mkdir -p ~/workingdirectory↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; cd ~/workingdirectory↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># disable upgrade of Assams</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; touch .assams↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># disable upgrade of Claident</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; touch .claident↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># disable upgrade of PEAR</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; touch .pear↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># disable upgrade of VSEARCH</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; touch .vsearch↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># disable upgrade of NCBI BLAST+</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; touch .blast↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># execute upgrade
&gt; wget https://www.claident.org/installClaident_Debian.sh↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; sh installClaident_Debian.sh↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># disable upgrade of sff_extract</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; touch .sffextract↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># disable upgrade of HMMer</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; touch .hmmer↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># disable upgrade of MAFFT</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; touch .mafft↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># disable upgrade of Metaxa</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; touch .metaxa↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># disable upgrade of ITSx</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; touch .itsx↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># execute upgrade
&gt; wget https://www.claident.org/installOptions_Debian.sh↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; sh installOptions_Debian.sh↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># disable upgrade of ``overall'' BLAST and taxonomy databases</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; touch .overall↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># execute upgrade
&gt; wget https://www.claident.org/installDB_Debian.sh↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; sh installDB_Debian.sh↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># disable upgrade of ``rdp'' reference database for chimera detection</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; touch .rdp↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># disable upgrade of ``unite'' reference databases for chimera detection</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; touch .unite↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># execute upgrade
&gt; wget https://www.claident.org/installUCHIMEDB_Debian.sh↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; sh installUCHIMEDB_Debian.sh↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; cd ..↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; rm -r workingdirectory↓</span>
            </span>
</p>
</div>
</section>
<section id="Ch0.S1.SS2" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">0.1.2 </span>Installing to non-default path</h4>

<div id="Ch0.S1.SS2.p1" class="ltx_para">
<p class="ltx_p">If you install the softwares based on the above procedure, the softwares will be installed to <span class="ltx_text ltx_font_typewriter">/usr/local</span>.
The executable commands will be installed to <span class="ltx_text ltx_font_typewriter">/usr/local/bin</span>.
You can change these install path for coexistence with the other programs such as older versions like below.</p>
</div>
<div id="Ch0.S1.SS2.p2" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; mkdir -p ~/workingdirectory↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; cd ~/workingdirectory↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; export PREFIX=install_path↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; wget https://www.claident.org/installClaident_Debian.sh↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; sh installClaident_Debian.sh↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; wget https://www.claident.org/installOptions_Debian.sh↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; sh installOptions_Debian.sh↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; wget https://www.claident.org/installDB_Debian.sh↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; sh installDB_Debian.sh↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; wget https://www.claident.org/installUCHIMEDB_Debian.sh↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; sh installUCHIMEDB_Debian.sh↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; cd ..↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; rm -r workingdirectory↓</span>
            </span>
</p>
</div>
<div id="Ch0.S1.SS2.p3" class="ltx_para">
<p class="ltx_p">In this case, the following commands need to be executed before analysis.</p>
</div>
<div id="Ch0.S1.SS2.p4" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; export PATH=install_path/bin:$PATH↓</span>
          </p>
</div>
<div id="Ch0.S1.SS2.p5" class="ltx_para">
<p class="ltx_p">You can omit above command if the above command is added to <span class="ltx_text ltx_font_typewriter">~/.bash_profile</span> or <span class="ltx_text ltx_font_typewriter">~/.bashrc</span>.</p>
</div>
</section>
<section id="Ch0.S1.SS3" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">0.1.3 </span>How to install multiple versions in a computer</h4>

<div id="Ch0.S1.SS3.p1" class="ltx_para">
<p class="ltx_p">If you install Claident and the other softwares to default install path of a computer to which Claident was already installed, all softwares will be overwritten.
As noted above, multiple versions of Claident can coexist if you install Claident to non-default path.
Note that a configuration file <span class="ltx_text ltx_font_typewriter">.claident</span> placed at a home directory of login user (<span class="ltx_text ltx_font_typewriter">/home/username</span>) or <span class="ltx_text ltx_font_typewriter">/etc/claident</span> cannot coexist at the same path.
You need to replace this file before changing the version of Claident.
The configuration file at the home directory of login user will be used preferentially.
To use multiple version, I recommend to make user account for each version and to install Claident to the home directory of each user.
Then, the version of Claident can be switched by switching login user.</p>
</div>
</section>
</section>
</section>
<section id="Ch1" class="ltx_chapter">
<h2 class="ltx_title ltx_title_chapter">
<span class="ltx_tag ltx_tag_chapter">Chapter 1 </span>Sequencing of multiple samples by next-generation sequencers</h2>

<div id="Ch1.p1" class="ltx_para">
<p class="ltx_p">In this chaper, I explain brief overview of tagged multiplex sequencing method by Roche GS series sequencers, Ion PGM and Illumina MiSeq.
These sequencers can read over-400bp contiguously and are suitable for metabarcoding and DNA barcoding.
Note that MiSeq requires concatenation of paired-end reads.
Therefore, PCR amplicons should be 500bp or shorter (400bp is recommended) in order to concatenate paired-end reads.
Forward and reverse reads can be analyzed separetely, but I cannot recommend such analysis because reverse reads are usually low quality.</p>
</div>
<div id="Ch1.p2" class="ltx_para">
<p class="ltx_p">The next-generation sequencers output extremely large amount of nucleotide sequences in single run.
Running costs of single run is much higher than Sanger method-based sequencers.
To use such sequencers efficiently, multiplex sequencing method was developed.
Multiplex identifier tag sequences are added to target sequences to identify the sample of origin, and the multiple tagged samples are mixed and sequenced in single run in this method.
This method can extremely reduce per-sample sequencing costs.
Multiplex identifier tag is also called as “barcode”.
However, nucleotide sequence for DNA barcoding is called as “barcode sequence”.
This is very confusing and “multiplex identifier tag” is too long.
Thus, I call multiplex identifier tag sequence as just “tag” in this text.
Please notice that tag is often called as “index”.</p>
</div>
<div id="Ch1.p3" class="ltx_para">
<p class="ltx_p">In the following analysis, chimera sequences constructed in PCR and erroneous sequences potentially causes misinterpretation of analysis results.
If multiple PCR replicates are prepared, tagged and sequenced separately, shared sequences among all replicates can be considered as nonchimeric and less erroneous.
This is because there are huge number of sequence combinations and joint points but no error sequence pattern is only one for one true sequence and nonchimeric and no error sequences likely to be observed at all replicates.
Program cannot remove chimeras and errors enough but we can expect that the combination of PCR replicates and program improves removal efficiency of chimeras and errors.
After removal of chimeras and errors, the number of sequences of PCR replicates can be summed up and used in subsequent analysis.</p>
</div>
<section id="Ch1.S1" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">1.1 </span>PCR using tag- and adapter-jointed-primers</h3>

<div id="Ch1.S1.p1" class="ltx_para">
<p class="ltx_p">In order to add tag to amplicon, PCR using tag-jointed primer is the easiest way.
This method requires a set of tag-jointed primers.
In addition, library preparation kits for next-generation sequencers usually presume that the adapter sequences specified by manufacturers are added to the both end of target sequences.
Thus, the following tag- and adapter-jointed primer is used for PCR.</p>
<p class="ltx_p ltx_align_left" style="border-left:0.5em solid #000000;padding:0.5em 1em 0.5em 1em;">
          <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">5' ― [adapter] ― [tag] ― [specific primer] ― 3'</span>
        </p>
<p class="ltx_p">If this kind of primers are used for the both forward and reverse primers, the following amplicon sequences will be constructed.</p>
<p class="ltx_p ltx_align_left" style="border-left:0.5em solid #000000;padding:0.5em 1em 0.5em 1em;">
          <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">5' ― [adapter-F] ― [tag-F] ― [specific primer-F] ― [target sequence] ― [specific primer-R (reverse complement)] ― [tag-R (reverse complement)] ― [adapter-R (reverse complement)] ― 3'</span>
        </p>
<p class="ltx_p">In the case of single-end read, tag-F leads specific primer-F and target sequence in the sequence data.</p>
</div>
<div id="Ch1.S1.p2" class="ltx_para">
<p class="ltx_p">The supplement of <cite class="ltx_cite"><a href="#bib.bib7" title="" class="ltx_ref">Hamady <span class="ltx_text ltx_font_italic">et al.</span> (2008)</a></cite> may be useful for picking tag sequences.
In the case of single-end sequencing, 3’-side tag is not required, and tagless primer can be used for PCR.
In the case of paired-end sequencing, single index (tag) can be applied, but dual index (tag) is recommended for detecting unlikely tag combinations which means that forward and reverse sequences are mispaired.</p>
</div>
<div id="Ch1.S1.p3" class="ltx_para">
<p class="ltx_p">Using above primer sets for PCR, primers anneal to templates in “Y”-formation, and the amplicon sequences which have tags and adapters for both ends will be constructed.
Then, the amplicon solutions are mixed in the same concentration and sequenced based on manufacturer’s protocol.
Spectrophotometer (including Nanodrop) is inappropreate for the measurement of the concentration of solution because measurement of dsDNA using spectrophotometer is likely to be affected by the other contaminants.
I recommend Qubit (ThermoFisher) for measurement of dsDNA concentration.
Quantitative PCR-based method can also be recommended but it’s expensive and more time-consuming.</p>
</div>
<div id="Ch1.S1.p4" class="ltx_para">
<p class="ltx_p">Primer annealing position sequence can also be used for recognizing the sample of origin.
Therefore, the sequences of multiple loci, for example plant <span class="ltx_text ltx_font_italic">rbcL</span> and <span class="ltx_text ltx_font_italic">matK</span>, from same sample set tagged by same tag set can be multiplexed and sequenced.
Of course, the sequences of multiple loci can also be recognized by themselves.
Smaller number of cycles and longer extension time were recommended for PCR.
Because the required amount of DNA for sequence sample preparation is not so high, the larger number of cycles of PCR amplification is not needed.
The larger number of cycles and shorter extension time generates more incompletely extended amplicon sequences and the incompletely extended amplicon sequences are re-extend using different template sequences in next cycle.
Such sequences are called as “chimeric DNA”.
Chimeric DNAs causes a discovery of non-existent novel species or a overestimation of species diversity.
To reduce chimeric DNA construction, using high-fidelity DNA polymerase such as Phusion (Finnzymes) or KOD (TOYOBO) is effective.
<cite class="ltx_cite"><a href="#bib.bib17" title="" class="ltx_ref">Stevens <span class="ltx_text ltx_font_italic">et al.</span> (2013)</a></cite> reported that slowing cooling-down from denaturation temperature to annealing temperature reduced chimeric DNA construction.
If your thermal cycler can change cooling speed, slowing cooling-down from denaturation temperature to annealing temperature can be recommended.
Chimeric DNA sequences can also be eliminated by computer programs after sequencing.
Because chimera removal by programs is incomplete and the nonchimeric sequences shrink, we cannot do better than reduce chimeric DNA construction.</p>
</div>
<div id="Ch1.S1.p5" class="ltx_para">
<p class="ltx_p">In the case of hardly amplifiable templates, using Ampdirect Plus (Shimadzu) for PCR buffer or crushing by homogenizer or beads before DNA extraction is recommended.
Deep freezing before crushing can also be recommended.
Removal of polyphenols or polysaccharides might be required if your sample contain those chemicals.
If PCR amplification using tag- and adapter-jointed-primers fail, try two-step PCR that consist of primary PCR (20–30 cycles) using primers without tags and adapters, purification of amplicons by ExoSAP-IT, and secondary PCR (5–10cycles) using amplicons of primary PCR as templates and tag- and adapter-jointed-primers.</p>
</div>
<section id="Ch1.S1.SS1" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">1.1.1 </span>Decreasing costs by interim adapters</h4>

<div id="Ch1.S1.SS1.p1" class="ltx_para">
<p class="ltx_p">Tag- and adaper-jointed-primers are very long and expensive.
In addition, we need to buy tag- and adaper-jointed-primers for each locus.
To reduce cost of tag- and adaper-jointed-primers, interim adapter-jointed primers and two-step PCR is useful.
The following primer set is used in primary PCR.</p>
<p class="ltx_p ltx_align_left" style="border-left:0.5em solid #000000;padding:0.5em 1em 0.5em 1em;">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">5' ― [interim adapter] ― [specific primer] ― 3'</span>
          </p>
<p class="ltx_p">This PCR product have interim adapter sequences at the both ends.
This PCR product is used as template in secondary PCR after purification.
The following primer set is used in secondary PCR.</p>
<p class="ltx_p ltx_align_left" style="border-left:0.5em solid #000000;padding:0.5em 1em 0.5em 1em;">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">5' ― [adapter specified by manufacturer] ― [tag] ― [interim adapter] ― 3'</span>
          </p>
<p class="ltx_p">This two-step PCR enables us to reuse secondary PCR primers.
However, this two-step PCR may increase PCR errors and PCR amplification biases, and decrease target sequence lengths.
Note that final PCR product is constructed as the following style.</p>
<p class="ltx_p ltx_align_left" style="border-left:0.5em solid #000000;padding:0.5em 1em 0.5em 1em;">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">5' ― [adapter-F specified by manufacturer] ― [tag-F] ― [interim adapter-F] ― [specific primer-F] ― [target sequence] ― [specific primer-R (reverse complement)] ― [interim adapter-R (reverse complement)] ― [tag-R (reverse complement)] ― [adapter-R (reverse complement) specified by manufacturer] ― 3'</span>
          </p>
<p class="ltx_p">Illumina’s multiplex sequencing method <cite class="ltx_cite"><a href="#bib.bib9" title="" class="ltx_ref">(Illumina corporation, 2013)</a></cite> using Nextera XT Index Kit is same as the above method.
In the dual-index paired-end sequencing based on this method, the first read start from behind of interim adapter-F (i.e. head of specific primer-F) to target sequence.
The second read start from behind of interim adapter-R and contains tag-R (index1) sequence.
The third read start from behind of adapter-F and contains tag-F (index2) sequence.
The last read start from ahead of interim adapter-R (i.e. tail of specific primer-R) to target sequence.
The first, second, third and last reads are saved as <span class="ltx_text ltx_font_typewriter">*_R1_*.fastq.gz</span>, <span class="ltx_text ltx_font_typewriter">*_R2_*.fastq.gz</span>, <span class="ltx_text ltx_font_typewriter">*_R3_*.fastq.gz</span> and <span class="ltx_text ltx_font_typewriter">*_R4_*.fastq.gz</span>, respectively.
The first, second and third reads are same strand, but last read is reverse strand.
Because the sequencing primers for the first and the last reads are targeting interim adapter-F and interim adapter-R, respectively, the first and the last reads contains the sequences of specific primer-F and specific primer-R, respectively.
Thus, the target sequences contained in the first and the last reads are shrinked.
If the length of the target sequence is 500 bp or longer, there might be no overlap and paired-end reads cannot be concatenate.
If specific primer-F and specific primer-R are used as sequencing primers for the first and the last reads, you can exclude the sequences of specific primer-F and specific primer-R from the first and the last reads.
However, the following quality improvement method by insertion of N cannot be applied in such case.</p>
</div>
</section>
<section id="Ch1.S1.SS2" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">1.1.2 </span>Quality improvement by insertion of N</h4>

<div id="Ch1.S1.SS2.p1" class="ltx_para">
<p class="ltx_p">On the Illumina platform, luminescence of syntheses of DNA on a flowcell is detected by optical sensor.
PCR amplicons of metagenomes are single locus and much more homogeneous than genome shotgun or RNA-seq library sequences.
In such case, neighboring sequences on a flowcell is difficult to distinguish one from the other.
In addition, if the nucleotide of the most sequences (especially first 12 nucleotides) are the same and nonluminescence, the Illumina platform sequencer will determined as failure and crash.
To avoid this problem, insertion of <span class="ltx_text ltx_font_typewriter">NNNNNN</span> between specific primer and interim adapter is effective.
<span class="ltx_text ltx_font_typewriter">NNNNNN</span> of the head of sequences enables sequencers to distinguish neighboring sequences and prevent black out, and the sequencing quality therefore will be improved <cite class="ltx_cite"><a href="#bib.bib15" title="" class="ltx_ref">(Nelson <span class="ltx_text ltx_font_italic">et al.</span>, 2014)</a></cite>.
The varied length of <span class="ltx_text ltx_font_typewriter">NNNNNN</span> causes artificial frameshift and also effective <cite class="ltx_cite"><a href="#bib.bib6" title="" class="ltx_ref">(Fadrosh <span class="ltx_text ltx_font_italic">et al.</span>, 2014)</a></cite>.
PhiX control can be reduced by using the above methods, and the application sequences will increase.</p>
</div>
</section>
</section>
</section>
<section id="Ch2" class="ltx_chapter">
<h2 class="ltx_title ltx_title_chapter">
<span class="ltx_tag ltx_tag_chapter">Chapter 2 </span>Preprocessing of nucleotide sequence data</h2>

<div id="Ch2.p1" class="ltx_para">
<p class="ltx_p">Roche GS series sequencers and Ion PGM output raw sequencing data as <span class="ltx_text ltx_font_typewriter">*.sff</span>.
Illumina platform sequencers output <span class="ltx_text ltx_font_typewriter">*.fastq</span> files.
In this chapter, the procedures of demultiplexing, quality-trimming and quality-filtering.
The <span class="ltx_text ltx_font_typewriter">clsplitseq</span> command of Claident is recommended for demultiplexing because the programs provided by manufacturer ignores the quality of tag positions.
The following commands should be executed in the terminal or console.
Fundamental knowledge of terminal operations is required.
If you are unfamiliar with terminal operations, you need to become understandable about the contents of appendix <a href="#A2" title="Appendix B Terminal command examples ‣ Matabarcoding and DNA barcoding for Ecologists" class="ltx_ref"><span class="ltx_text ltx_ref_tag">B</span></a>.</p>
</div>
<section id="Ch2.S1" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">2.1 </span>Importing sequence data deposited to SRA/DRA/ERA or demultiplexed FASTQ</h3>

<div id="Ch2.S1.p1" class="ltx_para">
<p class="ltx_p">Claident assumes <span class="ltx_text ltx_font_typewriter">SequenceID__RunID__TagID__PrimerID</span> for definition lines of sequences, and <span class="ltx_text ltx_font_typewriter">RunID__TagID__PrimerID</span> for file names (without extension).
Therefore, the sequence data deposited to SRA/DRA/ERA or demultiplexed FASTQ cannot be used as is.
The <span class="ltx_text ltx_font_typewriter">climportfastq</span> of Claident can convert such data.
If your data is paired-end, you need to concatenate and filter the sequences before conversion (see section <a href="#Ch2.S3.SS3" title="2.3.3 Concatenating forward and reverse sequences ‣ 2.3 For Illumina platform sequences ‣ Chapter 2 Preprocessing of nucleotide sequence data ‣ Matabarcoding and DNA barcoding for Ecologists" class="ltx_ref"><span class="ltx_text ltx_ref_tag">2.3.3</span></a>).
The following plain text file is required for conversion.</p>
</div>
<div id="Ch2.S1.p2" class="ltx_para">
<p class="ltx_p ltx_align_left" style="background-color:#E6E6E6;padding-top:12pt;padding-bottom:12pt;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| SequenceFileName1 RunID__TagID__PrimerID</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| SequenceFileName2 RunID__TagID__PrimerID</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| SequenceFileName3 RunID__TagID__PrimerID</span>
          </span>
</p>
</div>
<div id="Ch2.S1.p3" class="ltx_para">
<p class="ltx_p">Dummy RunID and PrimerID is acceptable.
PrimerID need to be the same among the sample used the same primer set.
TagID need to be different among the different sample files.
TagID can be the same as the sequence file name.</p>
</div>
<div id="Ch2.S1.p4" class="ltx_para">
<p class="ltx_p">After the above file was prepared, execute <span class="ltx_text ltx_font_typewriter">climportfastq</span> like the following and the above file should be given as an input file.</p>
</div>
<div id="Ch2.S1.p5" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; climportfastq \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
          </span>
</p>
</div>
<div id="Ch2.S1.p6" class="ltx_para">
<p class="ltx_p">Then, you can find converted files in the output folder.
If your sequence data is single-end, quality filtering explained in section <a href="#Ch2.S2.SS3" title="2.2.3 Trimming low quality tail and filtering low quality sequences ‣ 2.2 For Roche GS series sequencers and Ion PGM ‣ Chapter 2 Preprocessing of nucleotide sequence data ‣ Matabarcoding and DNA barcoding for Ecologists" class="ltx_ref"><span class="ltx_text ltx_ref_tag">2.2.3</span></a> is recommended.</p>
</div>
</section>
<section id="Ch2.S2" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">2.2 </span>For Roche GS series sequencers and Ion PGM</h3>

<section id="Ch2.S2.SS1" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">2.2.1 </span>Converting SFF to FASTQ</h4>

<div id="Ch2.S2.SS1.p1" class="ltx_para">
<p class="ltx_p">First of all, conversion of raw SFF format file to FASTQ file is needed like the following.</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; sff_extract -c inputfile(SFF)↓</span>
          </p>
<p class="ltx_p"><span class="ltx_text ltx_font_typewriter">-c</span> argument enables trimming of <span class="ltx_text ltx_font_typewriter">TCAG</span> at the head of sequences.
If you add <span class="ltx_text ltx_font_typewriter">TCAG</span> to the head of tag sequences, do not use this argument.
Assuming your SFF file name is <span class="ltx_text ltx_font_typewriter">HOGEHOGE.sff</span>, <span class="ltx_text ltx_font_typewriter">HOGEHOGE.fastq</span> will be saved as FASTQ file.
<span class="ltx_text ltx_font_typewriter">HOGEHOGE.xml</span> will also be generated, but this is not required.
The output sequences have tag sequences at the beginning, followed by primer-F and target sequences, and primer-R (reverse complement) at the end.
Note that all sequences are not completely read from the beginning to the end, the incomplete sequences are included.
The <span class="ltx_text ltx_font_typewriter">sff_extract</span> command is used in this book, but any other programs which can clip <span class="ltx_text ltx_font_typewriter">TCAG</span> at the beginning can be used.
If the SFF to FASTQ converter program cannot clip <span class="ltx_text ltx_font_typewriter">TCAG</span> at the beginning, adding <span class="ltx_text ltx_font_typewriter">TCAG</span> to the beginning of tag sequences to give to <span class="ltx_text ltx_font_typewriter">clsplitseq</span> also works well, but the quality values will be strictly checked.</p>
</div>
</section>
<section id="Ch2.S2.SS2" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">2.2.2 </span>Demultiplexing of sequences</h4>

<div id="Ch2.S2.SS2.p1" class="ltx_para">
<p class="ltx_p">The FASTQ file that contain the sequences from multiple samples need to be demultiplexed based on tag sequences and primer sequences before the subsequent analysis.
To do this process, a FASTA file which contain tag sequences and another FASTA file which contain primer-F sequences are required.</p>
<p class="ltx_p ltx_align_left" style="background-color:#E6E6E6;padding-top:12pt;padding-bottom:12pt;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| &gt;TagID</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| [tag sequence]</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| &gt;examplesample1</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| ACGTACGT</span>
            </span>
</p>
<p class="ltx_p ltx_align_left" style="background-color:#E6E6E6;padding-top:12pt;padding-bottom:12pt;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| &gt;PrimerID</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| [primer sequence]</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| &gt;exampleprimer1</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| ACGTACGTACGTACGTACGT</span>
            </span>
</p>
<p class="ltx_p">Degenerate codes of nucleotides are not allowed for tag sequences, but those are allowed for primer sequences.
Both of tag and primer FASTQ files can contain multiple sequences.
If you use interim adapter explained in section <a href="#Ch1.S1.SS1" title="1.1.1 Decreasing costs by interim adapters ‣ 1.1 PCR using tag- and adapter-jointed-primers ‣ Chapter 1 Sequencing of multiple samples by next-generation sequencers ‣ Matabarcoding and DNA barcoding for Ecologists" class="ltx_ref"><span class="ltx_text ltx_ref_tag">1.1.1</span></a>, primer sequences should be written like the following.</p>
<p class="ltx_p ltx_align_left" style="background-color:#E6E6E6;padding-top:12pt;padding-bottom:12pt;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| &gt;PrimerID</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| [interim adapter][primer sequence]</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| &gt;exampleprimer1</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| TGATACTCGATACGTACGTACGTACGTACGTACGT</span>
            </span>
</p>
<p class="ltx_p">Thus, the sequences between tag and target sequences should be written in primer FASTA file.</p>
</div>
<div id="Ch2.S2.SS2.p2" class="ltx_para">
<p class="ltx_p">All the above files are prepared, the following command can demultiplex nucleotide sequences to each sample FASTQ file.</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clsplitseq \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--runname=RunID \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--tagfile=TagSequenceFile \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--primerfile=PrimerSequenceFile \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minqualtag=27 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
            </span>
</p>
<p class="ltx_p">RunID must differ among different sequencing runs.
RunID is given by sequencer in many cases, you can use such sequencer generated RunID.
RunID is usually contained in sequence file name or sequence name in sequence file, but the naming rules are different among sequencing platforms.
Therefore, <span class="ltx_text ltx_font_typewriter">clsplitseq</span> requires RunID given by user.
<span class="ltx_text ltx_font_typewriter">--minqualtag</span> is an argument that specifies minimum quality threshold of tag position sequences.
If 1 or more lower quality nucleotide than this threshold value is contained by a sequence, such sequence will be omitted from output sequences.
27 for minimum quality threshold is proposed by <cite class="ltx_cite"><a href="#bib.bib11" title="" class="ltx_ref">Kunin <span class="ltx_text ltx_font_italic">et al.</span> (2010)</a></cite> for 3’-tail trimming of the sequences of Roche GS series sequencers.
The different value might be more suitable for the other sequencers.
In many cases, 30 is used for minimum quality threshold and can be recommended.</p>
</div>
<div id="Ch2.S2.SS2.p3" class="ltx_para">
<p class="ltx_p">If multiplex sequencing technique is not used, <span class="ltx_text ltx_font_typewriter">--tagfile</span> argument can be omitted.
However, just omit of <span class="ltx_text ltx_font_typewriter">--tagfile</span> generates incompatible FASTQ files for Claident.
In such case, you should add identifier (dummy is acceptable) of tag sequences using <span class="ltx_text ltx_font_typewriter">--indexname=TagID</span> argument.</p>
</div>
<div id="Ch2.S2.SS2.p4" class="ltx_para">
<p class="ltx_p">The tag and primer position sequences are trimmed from the output sequences.
Tag position sequence match is evaluated exactly and strictly.
There are no arguments to tolerate a mismatch.
Primer position sequence is aligned based on Needleman-Wunsch algorithm and evaluated allowing 14% of mismatches (the threshold can be changed).
The output files are named as <span class="ltx_text ltx_font_typewriter">RunID__TagID__PrimerID.fastq.gz</span> and saved in the output folder.
<span class="ltx_text ltx_font_typewriter">clsplitseq</span> can use multiple CPUs for faster processing.
If your computer have 4 CPU cores, 4 should be speficied for <span class="ltx_text ltx_font_typewriter">--numthreads</span> argument.
Note that operating system and/or writing speed of storage devices might limit processing speed.
By default, the output files are compressed by GZIP.
Therefore, decompression is required to read/write by incompatible programs with GZIP-compressed FASTQ files.
The commands of Claident used below can treat GZIP-compressed FASTQ files.</p>
</div>
<div id="Ch2.S2.SS2.p5" class="ltx_para">
<p class="ltx_p">Before submission of manuscripts, sequence data need to be deposited to public database such as DDBJ Sequence Read Archive (DRA).
GZIP-compressed FASTQ files in this step can be used for the data deposition.</p>
</div>
<section id="Ch2.S2.SS2.SSSx1" class="ltx_subsubsection">
<h5 class="ltx_title ltx_title_subsubsection">If you sequenced a number of samples by multiple sequencing runs</h5>

<div id="Ch2.S2.SS2.SSSx1.p1" class="ltx_para">
<p class="ltx_p">Multiple demultiplexing by <span class="ltx_text ltx_font_typewriter">clsplitseq</span> are required.
However, <span class="ltx_text ltx_font_typewriter">clsplitseq</span> cannot write already existing folder by default.
The secondary run of <span class="ltx_text ltx_font_typewriter">clsplitseq</span> requires <span class="ltx_text ltx_font_typewriter">--append</span> argument like below.</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clsplitseq \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--runname=RunID \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--tagfile=TagSequenceFile \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--primerfile=PrimerSequenceFile \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minqualtag=27 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile1 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clsplitseq \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--runname=RunID \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--tagfile=TagSequenceFile \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--primerfile=PrimerSequenceFile \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minqualtag=27 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--append \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile2 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
              </span>
</p>
</div>
</section>
<section id="Ch2.S2.SS2.SSSx2" class="ltx_subsubsection">
<h5 class="ltx_title ltx_title_subsubsection">If your tag sequence lengths are unequal</h5>

<div id="Ch2.S2.SS2.SSSx2.p1" class="ltx_para">
<p class="ltx_p"><span class="ltx_text ltx_font_typewriter">clsplitseq</span> assumes that all tag sequence lengths are equal for faster processing.
The unequal length tags must be splitted to multiple tag sequence files and multiple demultiplexing runs of <span class="ltx_text ltx_font_typewriter">clcplitseq</span> are required as the following.</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clsplitseq \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--runname=RunID \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--tagfile=TagSequenceFile1 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--primerfile=PrimerSequenceFile \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minqualtag=27 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clsplitseq \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--runname=RunID \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--tagfile=TagSequenceFile2 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--primerfile=PrimerSequenceFile \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minqualtag=27 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--append \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
              </span>
</p>
</div>
</section>
<section id="Ch2.S2.SS2.SSSx3" class="ltx_subsubsection">
<h5 class="ltx_title ltx_title_subsubsection">Recognition and elimination of reverse primer positions</h5>

<div id="Ch2.S2.SS2.SSSx3.p1" class="ltx_para">
<p class="ltx_p">In the above procedure, reverse primer position and subsequent sequences are not eliminated.
Reverse primer position and subsequent sequences are artificial and should be eliminated if possible.
To do so, reverse primer sequence file like the following is required.</p>
<p class="ltx_p ltx_align_left" style="background-color:#E6E6E6;padding-top:12pt;padding-bottom:12pt;">
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| &gt;PrimerID</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| [primer sequence]</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| &gt;exampleprimer1</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| TCAGTCAGTCAGTCAGTCAG</span>
              </span>
</p>
<p class="ltx_p">Multiple reverse primers can written in this file.
Note that the N-th reverse primer sequence is assumed to associate with the N-th forward primer sequence.
Therefore, the different number of primer sequences between forward and reverse primer sequence files causes an error.
If there are the samples whose forward or reverse primer sequence is same but the other primer sequence is different, both combinations of forward and reverse primer sequences need to be written as different primers in the files.</p>
</div>
<div id="Ch2.S2.SS2.SSSx3.p2" class="ltx_para">
<p class="ltx_p">After the preparation of the above file, perform <span class="ltx_text ltx_font_typewriter">clsplitseq</span> as the following.</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clsplitseq \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--runname=RunID \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--tagfile=TagSequenceFile \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--primerfile=ForwardPrimerSequenceFile \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--reverseprimerfile=ReversePrimerSequenceFile \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--reversecomplement \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minqualtag=27 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
              </span>
</p>
<p class="ltx_p">In this processing, reverse-complement sequence of reverse primer is searched based on Needleman-Wunsch algorithm allowing 15% (this value can be changed) of mismatches and reverse primer position and subsequent sequence is eliminated in addition to the above process.
If reverse-complement sequence of reverse primer is not found and the other requirement is fullfilled, the sequence will be saved to output file by default.
The <span class="ltx_text ltx_font_typewriter">--needreverseprimer</span> argument is required to filter out the sequence which does not contain reverse-complement sequence of reverse primer.</p>
</div>
</section>
</section>
<section id="Ch2.S2.SS3" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">2.2.3 </span>Trimming low quality tail and filtering low quality sequences</h4>

<div id="Ch2.S2.SS3.p1" class="ltx_para">
<p class="ltx_p">FASTQ sequences have read quality information.
The low quality 3’-tail can be trimmed and the low quality sequences can be filtered out based on the quality values.
The <span class="ltx_text ltx_font_typewriter">clfilterseq</span> command can perform such processing as the following.</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clfilterseq \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minqual=27 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minquallen=3 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minlen=350 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--maxlen=400 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--maxplowqual=0.1 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile↓</span>
            </span>
</p>
</div>
<div id="Ch2.S2.SS3.p2" class="ltx_para">
<p class="ltx_p">The values of <span class="ltx_text ltx_font_typewriter">--minqual</span> and <span class="ltx_text ltx_font_typewriter">--minquallen</span> indicate the minimum threshold of read quality value and size of sliding window, respectively.
The above command trims 3’-tail positions until 3 bp long sequence whose read quality is 27 or higher in all 3 positions are observed.
In addition, trimmed sequences shorter than <span class="ltx_text ltx_font_typewriter">--minlen</span> will be filtered out and trimmed sequences longer than <span class="ltx_text ltx_font_typewriter">--maxlen</span> will be trimmed to <span class="ltx_text ltx_font_typewriter">--maxlen</span>.
The remaining sequences containing <span class="ltx_text ltx_font_typewriter">--maxplowqual</span> or more rate of lower quality positions than <span class="ltx_text ltx_font_typewriter">--minqual</span> will also be filtered out.
The output is a file by default, but can be saved to the file in the new folder using <span class="ltx_text ltx_font_typewriter">--output=folder</span> argument.
The output file name is same as the input file name in this case.
If you want to save the output files to the existing folder, add <span class="ltx_text ltx_font_typewriter">--append</span> argument.</p>
</div>
<div id="Ch2.S2.SS3.p3" class="ltx_para">
<p class="ltx_p">If you want to apply <span class="ltx_text ltx_font_typewriter">clfilterseq</span> to the all files in the output folder of <span class="ltx_text ltx_font_typewriter">clsplitseq</span>, run the following command.</p>
</div>
<div id="Ch2.S2.SS3.p4" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; for f in OutputFolderOfclsplitseq/*.fastq.gz↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">do clfilterseq \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--output=folder \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--append \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minqual=27 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minquallen=3 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minlen=350 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--maxlen=400 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--maxplowqual=0.1 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">$f \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">done↓</span>
            </span>
</p>
</div>
</section>
</section>
<section id="Ch2.S3" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">2.3 </span>For Illumina platform sequences</h3>

<section id="Ch2.S3.SS1" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">2.3.1 </span>Converting from BCL to FASTQ</h4>

<div id="Ch2.S3.SS1.p1" class="ltx_para">
<p class="ltx_p">The analysis software of Illumina platform sequences can demultiplex sequencing reads, but ignores read quality of tag positions.
Therefore, the sequences have low quality tag positions possibly saved to demultiplexed FASTQ.
To filering out such sequences, pre-demultiplexed FASTQ files are required and can be converted from BCL files with the aid of bcl2fastq.
There are 1.x and 2.x series of bcl2fastq and both series can be used for Claident.
However, the sequencers may be compatible to either 1.x or 2.x, you need to select proper version.
Pre-demultiplexed FASTQ can be demultiplexed by <span class="ltx_text ltx_font_typewriter">clsplitseq</span> in Claident.
See appendix to install bcl2fastq.</p>
</div>
<div id="Ch2.S3.SS1.p2" class="ltx_para">
<p class="ltx_p">To convert BCL to FASTQ, run data folder (superjacent folder of BaseCalls folder) need to be copied to the PC installed bcl2fastq.
If there is <span class="ltx_text ltx_font_typewriter">SampleSheet.csv</span> in run data folder, this file must be renamed or deleted.</p>
</div>
<div id="Ch2.S3.SS1.p3" class="ltx_para">
<p class="ltx_p">In the case of bcl2fastq 1.x, the following commands make FASTQ files from BCL files of 8 bp dual indexed 300PE sequencing data.</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; cd RunDataFolder↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; configureBclToFastq.pl \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--fastq-cluster-count 0 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--use-bases-mask Y300n,Y8,Y8,Y300n \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--input-dir BaseCalls \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--output-dir outputfolder↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; make -j4↓</span>
            </span>
</p>
<p class="ltx_p">The <span class="ltx_text ltx_font_typewriter">--fastq-cluster-count 0</span> argument disable large output file splitting.
The <span class="ltx_text ltx_font_typewriter">--use-bases-mask Y300n,Y8,Y8,Y300n</span> is an argument to save forward 300 bp read (last base is trimmed), 8 bp index 1 (reverse-complement of tag-R), 8 bp index 2 (tag-F) and reverse 300 bp read (last base is trimmed) to <span class="ltx_text ltx_font_typewriter">*_R1_001.fastq.gz</span>, <span class="ltx_text ltx_font_typewriter">*_R2_001.fastq.gz</span>, <span class="ltx_text ltx_font_typewriter">*_R3_001.fastq.gz</span> and <span class="ltx_text ltx_font_typewriter">*_R4_001.fastq.gz</span>, respectively.
The value of <span class="ltx_text ltx_font_typewriter">--use-bases-mask</span> argument need to be changed for the other sequencing settings.
For 6 bp single indexed 250SE and 8 bp dual indexed 300SE sequencing data, <span class="ltx_text ltx_font_typewriter">--use-bases-mask Y250n,Y6</span> and <span class="ltx_text ltx_font_typewriter">--use-bases-mask Y300n,Y8,Y8</span> should be suitable, respectively.
<span class="ltx_text ltx_font_typewriter">make -j4</span> executes the conversion using 4 CPUs.
The output files will be compressed by GZIP.
The extension <span class="ltx_text ltx_font_typewriter">.gz</span> of output files indicates that the file is compressed by GZIP.
Claident is compliant with GZIP-compressed FASTQ files and decompression is not required.</p>
</div>
<div id="Ch2.S3.SS1.p4" class="ltx_para">
<p class="ltx_p">In the case of bcl2fastq 2.x, perform the following command.</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; bcl2fastq \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--processing-threads NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--use-bases-mask Y300n,Y8,Y8,Y300n \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--runfolder-dir RunDataFolder \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--output-dir outputfolder↓</span>
            </span>
</p>
<p class="ltx_p">The <span class="ltx_text ltx_font_typewriter">--processing-threads</span>, <span class="ltx_text ltx_font_typewriter">--use-bases-mask</span> and <span class="ltx_text ltx_font_typewriter">--runfolder-dir</span> indicate the number of processor used in conversion, masking option in common with 1.x and run data folder, respectively.</p>
</div>
</section>
<section id="Ch2.S3.SS2" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">2.3.2 </span>Demultiplexing of sequences</h4>

<div id="Ch2.S3.SS2.p1" class="ltx_para">
<p class="ltx_p">FASTA files containing tag (index) sequences and primer sequences like the following are needed for demultiplexing.
FASTA files containing secondary tag (index) sequences and reverse primer sequences are also required for paired-end sequencing data.</p>
<p class="ltx_p ltx_align_left" style="background-color:#E6E6E6;padding-top:12pt;padding-bottom:12pt;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| &gt;TagID</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| [tag sequence]</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| &gt;examplesample1</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| ACGTACGT</span>
            </span>
</p>
<p class="ltx_p ltx_align_left" style="background-color:#E6E6E6;padding-top:12pt;padding-bottom:12pt;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| &gt;PrimerID</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| [primer sequence]</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| &gt;exampleprimer1</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| ACGTACGTACGTACGTACGT</span>
            </span>
</p>
<p class="ltx_p">Degenerate code is not allowed for tag sequences, but can be used in primer sequences.
Multiple tags and primers can be written in the files, but the N-th reverse tag/primer sequence is assumed to associate with the N-th forward tag/primer sequence.
Therefore, the different number of tag/primer sequences between forward and reverse tag/primer sequence files causes an error.
If there are the samples whose forward or reverse tag/primer sequence is same but the other tag/primer sequence is different, both combinations of forward and reverse tag/primer sequences need to be written as different tags/primers in the files.
If you added <span class="ltx_text ltx_font_typewriter">N</span> in front of primer, <span class="ltx_text ltx_font_typewriter">N</span> need to be added in primer sequence.
If your <span class="ltx_text ltx_font_typewriter">N</span> length is unequal, only the longest <span class="ltx_text ltx_font_typewriter">N</span> should be written in the file.</p>
</div>
<div id="Ch2.S3.SS2.p2" class="ltx_para">
<p class="ltx_p">All the required files prepared, the following command demultiplex sequences to each sample file.</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clsplitseq \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--runname=RunID \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--index1file=Index1Sequence(tag-Rrevcomp)File \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--index2file=Index2Sequence(tag-F)File \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--primerfile=ForwardPrimerSequenceFile \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--reverseprimerfile=ReversePrimerSequenceFile \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minqualtag=30 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberofCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile1 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile2 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile3 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile4 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
            </span>
</p>
<p class="ltx_p">The input files should be specified in the order of <span class="ltx_text ltx_font_typewriter">*_R1_001.fastq.gz</span>, <span class="ltx_text ltx_font_typewriter">*_R2_001.fastq.gz</span>, <span class="ltx_text ltx_font_typewriter">*_R3_001.fastq.gz</span> and <span class="ltx_text ltx_font_typewriter">*_R4_001.fastq.gz</span>.
The <span class="ltx_text ltx_font_typewriter">--index1file</span> and <span class="ltx_text ltx_font_typewriter">--index2file</span> arguments requires the FASTA sequence files of index 1 (reverse-complement of tag-R) and index 2 (tag-F), respectively.
By default, the acceptable mismatches are 14% and 15% for forward and reverse primers, respectively.
If you added <span class="ltx_text ltx_font_typewriter">N</span> in front of primer, the <span class="ltx_text ltx_font_typewriter">--truncateN=enable</span> argument need to be given.
This argument enables exclusion of <span class="ltx_text ltx_font_typewriter">N</span> of primer and matched positions of sequences in calculation of the rate of mismatches.
Therefore, only the longest <span class="ltx_text ltx_font_typewriter">N</span> is required to find <span class="ltx_text ltx_font_typewriter">N</span>-added primer even if the length of <span class="ltx_text ltx_font_typewriter">N</span> is unequal.
After the processing, the number of sequences in demultiplexed files should be compared with those in demultiplexed files generated by Illumina softwares.
Correctly demultiplexed files should contain fewer sequences than demultiplexed files generated by Illumina softwares.
If you used specific primers for sequencing primers, forward and reverse sequences do not contain specific primer positions.
In such cases, <span class="ltx_text ltx_font_typewriter">--primerfile</span> and <span class="ltx_text ltx_font_typewriter">--reverseprimerfile</span> arguments are not required, but <span class="ltx_text ltx_font_typewriter">--primername=PrimerID</span> argument need to be given for converting sequence names as compliant with Claident.
Dummy PrimerID is acceptable but no PrimerID is not.</p>
</div>
<div id="Ch2.S3.SS2.p3" class="ltx_para">
<p class="ltx_p">If you do not perform multiplex sequencing using tag/index, <span class="ltx_text ltx_font_typewriter">--index1file</span> and <span class="ltx_text ltx_font_typewriter">--index2file</span> arguments are unneeded, but <span class="ltx_text ltx_font_typewriter">--indexname=TagID</span> argument must be given for converting sequence names as compliant with Claident.
Dummy TagID is acceptable but no TagID is not.</p>
</div>
<div id="Ch2.S3.SS2.p4" class="ltx_para">
<p class="ltx_p">After demultiplexing, <span class="ltx_text ltx_font_typewriter">RunID__TagID__PrimerID.forward.fastq.gz</span> and <span class="ltx_text ltx_font_typewriter">RunID__TagID__PrimerID.reverse.fastq.gz</span> will be generated.
These GZIP-compressed FASTQ files can be used for data deposition to sequence read archive sites such as DDBJ Sequence Read Archive (DRA).
In deposition process to DRA, it is required that the sequence lengths are equal or not.
Because primer position sequences that can be unequal lengths even if only one primer set was used are eliminated from demultiplexed sequence files, do not specify that the sequence lengths are equal.</p>
</div>
</section>
<section id="Ch2.S3.SS3" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">2.3.3 </span>Concatenating forward and reverse sequences</h4>

<section id="Ch2.S3.SS3.SSSx1" class="ltx_subsubsection">
<h5 class="ltx_title ltx_title_subsubsection">In the case of overlapped paired-end</h5>

<div id="Ch2.S3.SS3.SSSx1.p1" class="ltx_para">
<p class="ltx_p">The <span class="ltx_text ltx_font_typewriter">clconcatpair</span> command in Claident can be used for concatenating overlapped paired-end sequence data.
The <span class="ltx_text ltx_font_typewriter">clconcatpair</span> concatenate forward and reverse sequences based on overlap positions using VSEARCH by the following command.</p>
</div>
<div id="Ch2.S3.SS3.SSSx1.p2" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clconcatpair \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--mode=OVL \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfolder \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
              </span>
</p>
</div>
<div id="Ch2.S3.SS3.SSSx1.p3" class="ltx_para">
<p class="ltx_p">This command finds <span class="ltx_text ltx_font_typewriter">*.forward.fastq</span> and <span class="ltx_text ltx_font_typewriter">*.reverse.fastq</span> in inputfolder, and concatenate the pairs automatically.
GZIP-compressed <span class="ltx_text ltx_font_typewriter">.gz</span> and/or BZIP2-compressed <span class="ltx_text ltx_font_typewriter">.bz2</span> files are also be found and concatenated.
Concatenated sequence files will be generated as <span class="ltx_text ltx_font_typewriter">*.fastq.gz</span> in outputfolder.</p>
</div>
<div id="Ch2.S3.SS3.SSSx1.p4" class="ltx_para">
<p class="ltx_p">If input file names are not compliant with <span class="ltx_text ltx_font_typewriter">*.forward.fastq</span> and <span class="ltx_text ltx_font_typewriter">*.reverse.fastq</span>, the following command can be used for concatenating a pair of files.</p>
</div>
<div id="Ch2.S3.SS3.SSSx1.p5" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clconcatpair \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--mode=OVL \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile1 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile2 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile↓</span>
              </span>
</p>
</div>
<div id="Ch2.S3.SS3.SSSx1.p6" class="ltx_para">
<p class="ltx_p">The forward and reverse sequence FASTQ files should be given as inputfile1 and inputfile2, respectively.
Addition of <span class="ltx_text ltx_font_typewriter">.gz</span> or <span class="ltx_text ltx_font_typewriter">.bz2</span> is required for output file compression.</p>
</div>
</section>
<section id="Ch2.S3.SS3.SSSx2" class="ltx_subsubsection">
<h5 class="ltx_title ltx_title_subsubsection">In the case of non-overlapped paired-end</h5>

<div id="Ch2.S3.SS3.SSSx2.p1" class="ltx_para">
<p class="ltx_p">If there are no overlaps between forward and reverse sequences, quality-trimming and quality-filtering using <span class="ltx_text ltx_font_typewriter">clfilterseq</span> like the following should be performed at first.</p>
</div>
<div id="Ch2.S3.SS3.SSSx2.p2" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clfilterseq \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minqual=30 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minquallen=3 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minlen=100 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--maxplowqual=0.1 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile1 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile2 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
              </span>
</p>
</div>
<div id="Ch2.S3.SS3.SSSx2.p3" class="ltx_para">
<p class="ltx_p">The values of <span class="ltx_text ltx_font_typewriter">--minqual</span> and <span class="ltx_text ltx_font_typewriter">--minquallen</span> indicate the minimum threshold of read quality value and size of sliding window, respectively.
The above command trims 3’-tail positions until 3 bp long sequence whose read quality is 30 or higher in all 3 positions are observed.
In addition, trimmed sequences shorter than <span class="ltx_text ltx_font_typewriter">--minlen</span> will be filtered out.
The remaining sequences containing <span class="ltx_text ltx_font_typewriter">--maxplowqual</span> or more rate of lower quality positions than <span class="ltx_text ltx_font_typewriter">--minqual</span> will also be filtered out.
In this process, filtering out one of the sequence of a pair, the other sequence of the pair will also be filtered out.
The output will be generated as the same name files in outputfolder.
If you want to output to existing folder, you need to add <span class="ltx_text ltx_font_typewriter">--append</span> argument.
To apply the above command to all the pairs of <span class="ltx_text ltx_font_typewriter">*.forward.fastq</span> and <span class="ltx_text ltx_font_typewriter">*.reverse.fastq</span> in the current folder, execute the following commands.</p>
</div>
<div id="Ch2.S3.SS3.SSSx2.p4" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; for f in `ls *.forward.fastq.gz | grep -P -o '^[^\.]+'`↓</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">do clfilterseq \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minqual=30 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minquallen=3 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minlen=100 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--maxplowqual=0.1 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">$f.forward.fastq.gz \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">$f.reverse.fastq.gz \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">done↓</span>
              </span>
</p>
</div>
<div id="Ch2.S3.SS3.SSSx2.p5" class="ltx_para">
<p class="ltx_p">After the quality-trimming and quality-filtering like above, perform sequence concatenation with the aid of <span class="ltx_text ltx_font_typewriter">clconcatpair</span> like below.</p>
</div>
<div id="Ch2.S3.SS3.SSSx2.p6" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clconcatpair \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--mode=NON \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfolder \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
              </span>
</p>
</div>
<div id="Ch2.S3.SS3.SSSx2.p7" class="ltx_para">
<p class="ltx_p">In this process, the forward and reverse sequences like the following are assumed as input.</p>
</div>
<div id="Ch2.S3.SS3.SSSx2.p8" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border-left:0.5em solid #000000;padding:0.5em 1em 0.5em 1em;">
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">5' ― forward sequence ― 3'</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">5' ― reverse sequence ― 3'</span>
              </span>
</p>
</div>
<div id="Ch2.S3.SS3.SSSx2.p9" class="ltx_para">
<p class="ltx_p">The <span class="ltx_text ltx_font_typewriter">clconcatpair --mode=NON</span> command will concatenate these sequence pairs and make sequences like the following.</p>
</div>
<div id="Ch2.S3.SS3.SSSx2.p10" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border-left:0.5em solid #000000;padding:0.5em 1em 0.5em 1em;">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">5' ― reverse sequence (reverse-complement) ― ACGTACGTACGTACGT ― forward sequence ― 3'</span>
            </p>
</div>
<div id="Ch2.S3.SS3.SSSx2.p11" class="ltx_para">
<p class="ltx_p">Conduct removal of noisy and/or chimeric sequences in the same way as concatenated overlapped paired-end sequence data.
In the sequence clustering by <span class="ltx_text ltx_font_typewriter">clclassseqv</span> and the raw reads mapping to centroid sequences by <span class="ltx_text ltx_font_typewriter">clrecoverseqv</span>, add <span class="ltx_text ltx_font_typewriter">--paddinglen=16</span> argument.
The concatenated sequences like above causes overvaluation of sequence similarity because of artificial padding sequence <span class="ltx_text ltx_font_typewriter">ACGTACGTACGTACGT</span>.
The <span class="ltx_text ltx_font_typewriter">--paddinglen=16</span> argument will offset such overvaluation by exclusion of <span class="ltx_text ltx_font_typewriter">ACGTACGTACGTACGT</span> from sequence similarity calculation and cluster concatenated sequences based on correct sequence similarity.</p>
</div>
<div id="Ch2.S3.SS3.SSSx2.p12" class="ltx_para">
<p class="ltx_p">In the estimation of host organism, split concatenated sequences based on <span class="ltx_text ltx_font_typewriter">ACGTACGTACGTACGT</span>, assign taxonomy to forward and reverse sequences separately.
Then, merge 2 taxonomy (see section <a href="#Ch7.S3" title="7.3 Merging multiple taxonomic assignments based on consensus ‣ Chapter 7 Estimation of host organisms of nucleotide sequences (DNA barcoding) ‣ Matabarcoding and DNA barcoding for Ecologists" class="ltx_ref"><span class="ltx_text ltx_ref_tag">7.3</span></a>).
Generally speaking, forward sequences shows higher quality than reverse sequences, prefering forward sequence taxonomy is recommended if there is no <span class="ltx_text ltx_font_italic">a priori</span> infomations about identification power and variability of forward and reverse sequences.
Sequence division based on <span class="ltx_text ltx_font_typewriter">ACGTACGTACGTACGT</span> can be applied to the sequences by the following command.</p>
</div>
<div id="Ch2.S3.SS3.SSSx2.p13" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt;cldivseq \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--query=ACGTACGTACGTACGT \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile1 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile2</span>
              </span>
</p>
</div>
<div id="Ch2.S3.SS3.SSSx2.p14" class="ltx_para">
<p class="ltx_p">The outputfile1 and outputfile2 contain reverse-complement of reverse sequences and forward sequences, respectively.</p>
</div>
</section>
<section id="Ch2.S3.SS3.SSSx3" class="ltx_subsubsection">
<h5 class="ltx_title ltx_title_subsubsection">Concatenating overlapping paired-end sequences using PEAR</h5>

<div id="Ch2.S3.SS3.SSSx3.p1" class="ltx_para">
<p class="ltx_p">PEAR <cite class="ltx_cite"><a href="#bib.bib19" title="" class="ltx_ref">(Zhang <span class="ltx_text ltx_font_italic">et al.</span>, 2014)</a></cite> can also be used for concatenation of overlapped paired-end sequences.
The following command will concatenate the pairs of sequences.</p>
</div>
<div id="Ch2.S3.SS3.SSSx3.p2" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; pear \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">-p 0.0001 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">-u 0 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">-j NumberOfCPUs \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">-f RunID__TagID__PrimerID.forward.fastq.gz \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">-r RunID__TagID__PrimerID.reverse.fastq.gz \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">-o RunID__TagID__PrimerID↓</span>
              </span>
</p>
</div>
<div id="Ch2.S3.SS3.SSSx3.p3" class="ltx_para">
<p class="ltx_p">If the processes correctly finished, the following files will be generated.</p>
</div>
<div id="Ch2.S3.SS3.SSSx3.p4" class="ltx_para">
<dl id="Ch2.I1" class="ltx_description">
<dt id="Ch2.I1.ix1"><span class="ltx_tag ltx_tag_description">
                  <span class="ltx_text" style="font-size:90%;">RunID__TagID__PrimerID.assembled.fastq</span>
                </span></dt>
<dd>
<div id="Ch2.I1.ix1.p1" class="ltx_para">
<p class="ltx_p">
                    <span class="ltx_text" style="font-size:90%;">Concatenated sequences.</span>
                  </p>
</div>
</dd>
<dt id="Ch2.I1.ix2"><span class="ltx_tag ltx_tag_description">
                  <span class="ltx_text" style="font-size:90%;">RunID__TagID__PrimerID.unassembled.forward.fastq</span>
                </span></dt>
<dd>
<div id="Ch2.I1.ix2.p1" class="ltx_para">
<p class="ltx_p">
                    <span class="ltx_text" style="font-size:90%;">Unconcatenated forward sequences.</span>
                  </p>
</div>
</dd>
<dt id="Ch2.I1.ix3"><span class="ltx_tag ltx_tag_description">
                  <span class="ltx_text" style="font-size:90%;">RunID__TagID__PrimerID.unassembled.reverse.fastq</span>
                </span></dt>
<dd>
<div id="Ch2.I1.ix3.p1" class="ltx_para">
<p class="ltx_p">
                    <span class="ltx_text" style="font-size:90%;">Unconcatenated reverse sequences.</span>
                  </p>
</div>
</dd>
<dt id="Ch2.I1.ix4"><span class="ltx_tag ltx_tag_description">
                  <span class="ltx_text" style="font-size:90%;">RunID__TagID__PrimerID.discarded.fastq</span>
                </span></dt>
<dd>
<div id="Ch2.I1.ix4.p1" class="ltx_para">
<p class="ltx_p">
                    <span class="ltx_text" style="font-size:90%;">Discarded sequences by statistical test.</span>
                  </p>
</div>
</dd>
</dl>
</div>
<div id="Ch2.S3.SS3.SSSx3.p5" class="ltx_para">
<p class="ltx_p">Only <span class="ltx_text ltx_font_typewriter">RunID__TagID__PrimerID.assembled.fastq</span> is required in subsequent procedures.
These output files are not compressed and consume large amount of storages, compression by GZIP or BZIP2 is recommended.</p>
</div>
<div id="Ch2.S3.SS3.SSSx3.p6" class="ltx_para">
<p class="ltx_p">To apply concatenation to all the pairs of forward and reverse sequence files by PEAR, execute the following command.</p>
</div>
<div id="Ch2.S3.SS3.SSSx3.p7" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; for f in `ls *.forward.fastq.gz | grep -P -o '^[^\.]+'`↓</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">do pear \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">-p 0.0001 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">-u 0 \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">-j NumberOfCPUs \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">-f $f.forward.fastq.gz \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">-r $f.reverse.fastq.gz \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">-o $f↓</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">done↓</span>
              </span>
</p>
</div>
</section>
<section id="Ch2.S3.SS3.SSSx4" class="ltx_subsubsection">
<h5 class="ltx_title ltx_title_subsubsection">Concatenating overlapping paired-end sequences using VSEARCH</h5>

<div id="Ch2.S3.SS3.SSSx4.p1" class="ltx_para">
<p class="ltx_p">VSEARCH can also be used directly for concatenation of overlapped paired-end sequences.
The following command will concatenate the pairs of sequences.</p>
</div>
<div id="Ch2.S3.SS3.SSSx4.p2" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; vsearch \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--threads NumberOfCPUs \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--fastq_mergepairs RunID__TagID__PrimerID.forward.fastq.gz \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--reverse RunID__TagID__PrimerID.reverse.fastq.gz \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--fastq_allowmergestagger \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--fastqout RunID__TagID__PrimerID.assembled.fastq↓</span>
              </span>
</p>
</div>
<div id="Ch2.S3.SS3.SSSx4.p3" class="ltx_para">
<p class="ltx_p">If the amplicon sequences are shorter than read length, read tail of forward sequence possibly exceeds read head of reverse sequence or read tail of reverse sequence possibly exceeds read head of forward sequence.
VSEARCH does not concatenate such sequences by default.
The <span class="ltx_text ltx_font_typewriter">--fastq_allowmergestagger</span> argument enables such sequence concatenation.
The overhang positions which exceeded read head of the other sequence will be eliminated because such positions are artificial.
The <span class="ltx_text ltx_font_typewriter">--fastq_allowmergestagger</span> argument is not required if there is no such sequences.
Using PEAR, the same processing as VSEARCH with <span class="ltx_text ltx_font_typewriter">--fastq_allowmergestagger</span> will be performed.
The unconcatenated forward and reverse sequences can be obtained by <span class="ltx_text ltx_font_typewriter">--fastqout_notmerged_fwd outputfile</span> and <span class="ltx_text ltx_font_typewriter">--fastqout_notmerged_rev outputfile</span> arguments, respectively.
The minimum overlap length, the minimum length of concatenated sequence, the maximum length of concatenated sequence, the maximum number of allowed mismatches and the maximum allowed expected errors in concatenated sequence can be specified by <span class="ltx_text ltx_font_typewriter">--fastq_minovlen</span>, <span class="ltx_text ltx_font_typewriter">--fastq_minmergelen</span>, <span class="ltx_text ltx_font_typewriter">--fastq_maxmergelen</span>, <span class="ltx_text ltx_font_typewriter">--fastq_maxdiffs</span> and <span class="ltx_text ltx_font_typewriter">--fastq_maxee</span>, respectively.
In the concatenation of overlapped paired-end sequences by <span class="ltx_text ltx_font_typewriter">clconcatpair</span>, <span class="ltx_text ltx_font_typewriter">--fastq_minovlen 20 --fastq_maxdiffs 20</span> is used by default, but <span class="ltx_text ltx_font_typewriter">--fastq_minovlen 10 --fastq_maxdiffs 5</span> is used by default of VSEARCH.</p>
</div>
<div id="Ch2.S3.SS3.SSSx4.p4" class="ltx_para">
<p class="ltx_p">To apply concatenation by VSEARCH to all the files in current folder, execute the following command.</p>
</div>
<div id="Ch2.S3.SS3.SSSx4.p5" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; for f in `ls *.forward.fastq.gz | grep -P -o '^[^\.]+'`↓</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">do vsearch \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--threads NumberOfCPUs \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--fastq_mergepairs $f.forward.fastq.gz \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--reverse $f.reverse.fastq.gz \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--fastq_allowmergestagger \</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--fastqout $f.assembled.fastq↓</span>
              </span>
<span class="ltx_p ltx_align_left">
                <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">done↓</span>
              </span>
</p>
</div>
</section>
</section>
<section id="Ch2.S3.SS4" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">2.3.4 </span>Filtering potentially erroneous sequences</h4>

<div id="Ch2.S3.SS4.p1" class="ltx_para">
<p class="ltx_p">There are read quality values in the FASTQ files.
Therefore, we can filter out potentially erroneous sequences using these quality values.
To do so, the following command can conduct such processing.</p>
</div>
<div id="Ch2.S3.SS4.p2" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clfilterseq \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minqual=30 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--maxplowqual=0.1 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile↓</span>
            </span>
</p>
</div>
<div id="Ch2.S3.SS4.p3" class="ltx_para">
<p class="ltx_p">The sequences containing <span class="ltx_text ltx_font_typewriter">--maxplowqual</span> or more rate of lower quality positions than <span class="ltx_text ltx_font_typewriter">--minqual</span> will also be filtered out by the above command.
The output is a file by default, but adding <span class="ltx_text ltx_font_typewriter">--output=folder</span> argument changes to save output as the same name file in the outputfolder.
If you want to save output file to existing folder, <span class="ltx_text ltx_font_typewriter">--append</span> argument is needed.
In the case of concatenated sequences of overlapped paired-end sequences generated by Illumina platform sequencers, positions close to the both end is usually high quality and overlapped positions is also high quality if the same positions of forward and reverse sequences are matched.
Therefore, trimming low quality positions close to the both end is needless.
Filtering out sequences containing low quality positions is recommended for concatenated overlapped paired-end sequences.
The existing sequence filtering programs such as FastQC <cite class="ltx_cite"><a href="#bib.bib2" title="" class="ltx_ref">(Andrews, 2010)</a></cite> or PRINSEQ <cite class="ltx_cite"><a href="#bib.bib16" title="" class="ltx_ref">(Schmieder &amp; Edwards, 2011)</a></cite> are also recommended.</p>
</div>
<div id="Ch2.S3.SS4.p4" class="ltx_para">
<p class="ltx_p">To apply the same processing to the concatenated sequences by PEAR, execute the following command.</p>
</div>
<div id="Ch2.S3.SS4.p5" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; for f in *.assembled.fastq↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">do clfilterseq \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--output=folder \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--append \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minqual=30 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--maxplowqual=0.1 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">$f \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">done↓</span>
            </span>
</p>
</div>
<div id="Ch2.S3.SS4.p6" class="ltx_para">
<p class="ltx_p">Using quality values, we can calculate expected number of read errors.
Quality-filtering based on the maximum allowed expected errors in input sequences can also be applied using VSEARCH.
The following command can apply such quality-filtering to the input sequences.</p>
</div>
<div id="Ch2.S3.SS4.p7" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; vsearch \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--threads NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--fastq_filter inputfile \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--fastq_maxee 1.0 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--fastqout outputfile↓</span>
            </span>
</p>
</div>
<div id="Ch2.S3.SS4.p8" class="ltx_para">
<p class="ltx_p">For the single-end sequence data and unconcatenated sequence data, same quality-trimming and quality-filtering as Roche GS series sequencers and Ion PGM can be recommended (see section <a href="#Ch2.S2.SS3" title="2.2.3 Trimming low quality tail and filtering low quality sequences ‣ 2.2 For Roche GS series sequencers and Ion PGM ‣ Chapter 2 Preprocessing of nucleotide sequence data ‣ Matabarcoding and DNA barcoding for Ecologists" class="ltx_ref"><span class="ltx_text ltx_ref_tag">2.2.3</span></a>).
Note that thresholds for quality values and read lengths should be changed.</p>
</div>
</section>
</section>
<section id="Ch2.S4" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">2.4 </span>If you sequenced same PCR amplicons multiply or replicated PCR amplicons of same templates</h3>

<div id="Ch2.S4.p1" class="ltx_para">
<p class="ltx_p">Because chimeric sequences are constructed in each PCR tube, whether the sequences are came from same tube or not should be given to analysis programs.
Therefore, several procedure is required to give such information to the programs in the cases of replicated sequencing of same PCR amplicons and sequencing of replicated PCR amplicons of same templates.</p>
</div>
<section id="Ch2.S4.SS1" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">2.4.1 </span>In the case of sequencing of replicated PCR amplicons of same templates using same tags in the same run</h4>

<div id="Ch2.S4.SS1.p1" class="ltx_para">
<p class="ltx_p">In this case, chimera removal based on replicates of PCR cannot be applied.
This case can treat as same as unreplicated PCR.</p>
</div>
</section>
<section id="Ch2.S4.SS2" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">2.4.2 </span>In the case of sequencing of replicated PCR amplicons of same templates using different tags in the same run</h4>

<div id="Ch2.S4.SS2.p1" class="ltx_para">
<p class="ltx_p">In Claident, RunID__TagID__PrimerID is used as sample IDs.
Therefore, there are multiple samples from the same templates in this case.
It may be good idea that common sequences among replicated samples are treated as noiseless and nonchimeric, and uncommon sequences are treated as noisy and/or chimeric sequences if noise occurrence and chimera formation can be assumed as random.
However, noise occurrence and chimera formation likely to be nonrandom.
Noisy and/or chimeric sequences might be occurred across all replicates.
The effectiveness of this method does not confirmed enough, and combination of noisy and/or chimeric sequence removal based on PCR replicates and algorithms can be recommended (see also <cite class="ltx_cite"><a href="#bib.bib12" title="" class="ltx_ref">Lange <span class="ltx_text ltx_font_italic">et al.</span> (2015)</a></cite>).
Such combination is supported in Claident, it is explained later.
One of the final output of Claident is a sample x OTU table containing the number of sequences in every cell.
The table modification command <span class="ltx_text ltx_font_typewriter">clfiltersum</span> can be used for integration of multiple samples to one, and cell numbers will be summed up.</p>
</div>
</section>
<section id="Ch2.S4.SS3" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">2.4.3 </span>In the case of sequencing of same PCR amplicons using same tags in different runs</h4>

<div id="Ch2.S4.SS3.p1" class="ltx_para">
<p class="ltx_p">In Claident, RunID__TagID__PrimerID is used as sample IDs.
Therefore, there are multiple samples from the same templates in this case.
Such samples can be treated as separate or integrated to one sample.
It is recommended that separate samples have been used subsequent analysis without any change and such samples are finally integrated in a sample x OTU table.
The table modification command <span class="ltx_text ltx_font_typewriter">clfiltersum</span> can be used for integration of multiple samples to one, and cell numbers will be summed up.</p>
</div>
<div id="Ch2.S4.SS3.p2" class="ltx_para">
<p class="ltx_p">If you want to integrate multiple samples from the same templates at this time, execute the following commands.</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clsplitseq \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">*snip* \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--runname=FOO \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile1 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clsplitseq \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">*snip* \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--runname=FOO \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--append \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile2 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
            </span>
</p>
<p class="ltx_p">The inputfile1 and inputfile2 are the FASTQ files of first and second run, respectively.
The <span class="ltx_text ltx_font_typewriter">--runname</span> argument replaces RunID of the sequence names to <span class="ltx_text ltx_font_typewriter">FOO</span>.
Thus, all the sequences of both input files will be saved to <span class="ltx_text ltx_font_typewriter">FOO__TagID__PrimerID.fastq.gz</span> in the output folder.</p>
</div>
<div id="Ch2.S4.SS3.p3" class="ltx_para">
<p class="ltx_p">In Claident, the sequences whose names contain the same RunID, TagID and PrimerID are treated as the sequences from the same samples.
If tag and primer sequence files are the same, TagID and PrimerID of output sequences are the same.
Replacing RunID to <span class="ltx_text ltx_font_typewriter">FOO</span>, all of RunID, TagID and PrimerID become the same in the sequences from the same samples.</p>
</div>
</section>
<section id="Ch2.S4.SS4" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">2.4.4 </span>In the case of sequencing of replicated PCR amplicons of same templates using same tags in different runs</h4>

<div id="Ch2.S4.SS4.p1" class="ltx_para">
<p class="ltx_p">You should have multiple FASTQ files.
Run demultiplexing by <span class="ltx_text ltx_font_typewriter">clsplitseq</span> and quality-trimming and quality-filtering by <span class="ltx_text ltx_font_typewriter">clfilterseq</span> separately, and save processed sequences to different folders.
After separate processing of noisy and/or chimeric sequence removal, give all processed sequence files to clustering programs at once.
One of the final output of Claident is a sample x OTU table containing the number of sequences in every cell.
The table modification command <span class="ltx_text ltx_font_typewriter">clfiltersum</span> can be used for integration of multiple samples to one, and cell numbers will be summed up.</p>
</div>
</section>
<section id="Ch2.S4.SS5" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">2.4.5 </span>In the case of sequencing of replicated PCR amplicons of same templates using different tags in different runs</h4>

<div id="Ch2.S4.SS5.p1" class="ltx_para">
<p class="ltx_p">In Claident, RunID__TagID__PrimerID is used as sample IDs.
Therefore, there are multiple replicated samples from the same templates in this case.
Such samples can be treated as separate or integrated to one sample.
It is recommended that separate samples have been used subsequent analysis without any change and such samples are finally integrated in a sample x OTU table.
It may be good idea that common sequences among replicated samples are treated as noiseless and nonchimeric, and uncommon sequences are treated as noisy and/or chimeric sequences if noise occurrence and chimera formation can be assumed as random.
Noisy and/or chimeric sequences might be occurred across all replicates.
The effectiveness of this method does not confirmed enough, and combination of noisy and/or chimeric sequence removal based on PCR replicates and algorithms can be recommended (see also <cite class="ltx_cite"><a href="#bib.bib12" title="" class="ltx_ref">Lange <span class="ltx_text ltx_font_italic">et al.</span> (2015)</a></cite>).
Such combination is supported in Claident, it is explained later.
One of the final output of Claident is a sample x OTU table containing the number of sequences in every cell.
The table modification command <span class="ltx_text ltx_font_typewriter">clfiltersum</span> can be used for integration of multiple samples to one, and cell numbers will be summed up.</p>
</div>
<div id="Ch2.S4.SS5.p2" class="ltx_para">
<p class="ltx_p">In the demultiplexing using <span class="ltx_text ltx_font_typewriter">clsplitseq</span>, process multiple FASTQ files like the following.</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clsplitseq \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">*snip* \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--tagfile=TagSequenceFile1 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--primerfile=PrimerSequenceFile \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile1 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clsplitseq \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">*snip* \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--tagfile=TagSequenceFile2 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--primerfile=PrimerSequenceFile \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--append \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile2 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
            </span>
</p>
<p class="ltx_p">The sequence file of primer and the output folder should be the same.
The different RunID should be given for <span class="ltx_text ltx_font_typewriter">--runname</span> argument.
The different tag sequence files should be prepared and given.
Note that the same TagID should be specified for the replicated samples from the same templates even if tag/index sequence is different.
In addition, different TagID should be specified for the samples from different templates even if tag/index sequence is the same.
In the case of the following tag sequence files, the sequences of sample1 were added <span class="ltx_text ltx_font_typewriter">ACGTACGT</span> as a tag in the first run, and <span class="ltx_text ltx_font_typewriter">ATGCATGC</span> in the second run.</p>
<p class="ltx_p ltx_align_left" style="background-color:#E6E6E6;padding-top:12pt;padding-bottom:12pt;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| &gt;sample1</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| ACGTACGT</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| &gt;sample2</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| ATGCATGC</span>
            </span>
</p>
<p class="ltx_p ltx_align_left" style="background-color:#E6E6E6;padding-top:12pt;padding-bottom:12pt;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| &gt;sample1</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| ATGCATGC</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| &gt;sample2</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| ACGTACGT</span>
            </span>
</p>
<p class="ltx_p">The sequences of sample2 were added <span class="ltx_text ltx_font_typewriter">ATGCATGC</span> as a tag in the first run, and <span class="ltx_text ltx_font_typewriter">ACGTACGT</span> in the second run.
The subsequent analysis is explained later.</p>
</div>
</section>
</section>
</section>
<section id="Ch3" class="ltx_chapter">
<h2 class="ltx_title ltx_title_chapter">
<span class="ltx_tag ltx_tag_chapter">Chapter 3 </span>Noisy and/or chimeric sequence removal</h2>

<div id="Ch3.p1" class="ltx_para">
<p class="ltx_p">Claident can detect noisy sequences containing read errors and/or copy errors based on sequence abandance.
This method is similar to the method implemented in CD-HIT-OTU <cite class="ltx_cite"><a href="#bib.bib14" title="" class="ltx_ref">(Li <span class="ltx_text ltx_font_italic">et al.</span>, 2012)</a></cite>.
In the old pipeline using Assams for dereplication and clustering, chimera removal based on UCHIME <cite class="ltx_cite"><a href="#bib.bib5" title="" class="ltx_ref">(Edgar <span class="ltx_text ltx_font_italic">et al.</span>, 2011)</a></cite> algorithm can be applied.
Chimera removal based on UCHIME <cite class="ltx_cite"><a href="#bib.bib5" title="" class="ltx_ref">(Edgar <span class="ltx_text ltx_font_italic">et al.</span>, 2011)</a></cite> algorithm is applied after OTU picking in the new pipeline using VSEARCH for dereplication and clustering.</p>
</div>
<section id="Ch3.S1" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">3.1 </span>New pipeline using VSEARCH</h3>

<div id="Ch3.S1.p1" class="ltx_para">
<p class="ltx_p">Run the following command to perform noisy sequence detection and removal.
Note that multiple input files can be given, and all sequence files of the same run should be given at once because sequencing quality varied among different runs.
If there are too many sequences and the processing requires long time, give one sequence file and run the following command several times.</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clcleanseqv \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--derepmode=PREFIX \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--primarymaxnmismatch=0 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--secondarymaxnmismatch=1 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--pnoisycluster=0.5 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile1 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">中略 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfileN \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
          </span>
</p>
<p class="ltx_p">Whether full-length perfect matching (<span class="ltx_text ltx_font_typewriter">FULLLENGTH</span>) or prefix search (<span class="ltx_text ltx_font_typewriter">PREFIX</span>) applied in dereplication is given for the <span class="ltx_text ltx_font_typewriter">--derepmode</span> argument.
<span class="ltx_text ltx_font_typewriter">FULLLENGTH</span> is recommended for concatenated sequences of overlapped paired-end sequencing.
<span class="ltx_text ltx_font_typewriter">PREFIX</span> is recommended for single-end or concatenated sequences of non-overlapped paired-end sequencing.
The <span class="ltx_text ltx_font_typewriter">--primarymaxnmismatch</span> argument indicates the number of mismatches in primary clustering and 0 is recommended for the most cases.
The <span class="ltx_text ltx_font_typewriter">--secondarymaxnmismatch</span> argument describes the number of mismatches in secondary clustering and 1 is recommended for the most cases.
For the noisy data, use <span class="ltx_text ltx_font_typewriter">--primarymaxnmismatch=1 --secondarymaxnmismatch=3</span> or <span class="ltx_text ltx_font_typewriter">--primarymaxnmismatch=2 --secondarymaxnmismatch=5</span>.
Twice as <span class="ltx_text ltx_font_typewriter">--primarymaxnmismatch</span> plus one should be specified for <span class="ltx_text ltx_font_typewriter">--secondarymaxnmismatch</span>.
The <span class="ltx_text ltx_font_typewriter">--pnoisycluster</span> argument determines sensitivity of noise detection.
Decimal value larger than 0 and smaller than 1 must be specified for this argument.
The larger value acquire higher sensitivity.
If you use 97% identity cutoff in clustering, 0.5 is recommended and this is the default value.
If you use 99% or larger identity cutoff in clustering, 0.9 or larger value might be more suitable.
The larger value causes exclusion of more low abundance sequences, more sequences should be obtained per sample.</p>
</div>
<div id="Ch3.S1.p2" class="ltx_para">
<p class="ltx_p">The following files should be saved in the output folder.</p>
<dl id="Ch3.I1" class="ltx_description">
<dt id="Ch3.I1.ix1"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">parameter.txt</span>
            </span></dt>
<dd>
<div id="Ch3.I1.ix1.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">the minimum size of primary clusters remained</span>
              </p>
</div>
</dd>
<dt id="Ch3.I1.ix2"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">primarycluster.denoised.fasta.gz</span>
            </span></dt>
<dd>
<div id="Ch3.I1.ix2.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">representative sequences of primary clusters determined as non-noisy</span>
              </p>
</div>
</dd>
<dt id="Ch3.I1.ix3"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">primarycluster.fasta.gz</span>
            </span></dt>
<dd>
<div id="Ch3.I1.ix3.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">representative sequences of primary clusters</span>
              </p>
</div>
</dd>
<dt id="Ch3.I1.ix4"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">secondarycluster.fasta.gz</span>
            </span></dt>
<dd>
<div id="Ch3.I1.ix4.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">representative sequences of secondary clusters</span>
              </p>
</div>
</dd>
<dt id="Ch3.I1.ix5"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">RunID__TagID__PrimerID.noisyreads.txt.gz</span>
            </span></dt>
<dd>
<div id="Ch3.I1.ix5.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">list of sequences determined as noisy</span>
              </p>
</div>
</dd>
<dt id="Ch3.I1.ix6"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">RunID__TagID__PrimerID.singletons.txt.gz</span>
            </span></dt>
<dd>
<div id="Ch3.I1.ix6.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">list of singletons after primary clustering</span>
              </p>
</div>
</dd>
</dl>
<p class="ltx_p">Many other files might be generated and do not delete such files because those might be required in subsequent analysis.</p>
</div>
<section id="Ch3.S1.SS1" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">3.1.1 </span>Noisy and/or chimeric sequence removal based on PCR replicates</h4>

<div id="Ch3.S1.SS1.p1" class="ltx_para">
<p class="ltx_p">To detect and remove noisy and/or chimeric sequences using PCR replicates, which samples are from the same templates need to be provided as a text file like below.</p>
<p class="ltx_p ltx_align_left" style="background-color:#E6E6E6;padding-top:12pt;padding-bottom:12pt;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| sample1  sample2  sample3</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| sample4  sample5</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| sample6  sample7</span>
            </span>
</p>
<p class="ltx_p">Tab-delimited sample list in a line indicates samples from the same templates.
Samples from different templates must be placed in different lines.
3 or more replicates are allowed.
The number of replicates can vary among templates.</p>
</div>
<div id="Ch3.S1.SS1.p2" class="ltx_para">
<p class="ltx_p">The above file are prepared, the following command removes uncommon primary cluster as noisy and/or chimeric.</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clcleanseqv \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--replicatelist=ListOfPCRreplicates \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--derepmode=PREFIX \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--primarymaxnmismatch=0 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--secondarymaxnmismatch=1 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--pnoisycluster=0.5 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile1 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">*snip* \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfileN \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
            </span>
</p>
<p class="ltx_p">If 3 or more replicates are available, only common sequences among all replicates are determined as non-noisy and nonchimeric.
If a primary cluster occured among multiple templates and the cluster determined as noisy or chimeric in a template, the cluster determined as noisy or chimeric in the other templates and will be excluded from all the samples by default.
However, the following arguments can change this decision method.</p>
<dl id="Ch3.I2" class="ltx_description">
<dt id="Ch3.I2.ix1"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minnreplicate</span>
              </span></dt>
<dd>
<div id="Ch3.I2.ix1.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">Specified as integer larger than 1.
If the number of replicates occured is equal to or larger than this value, such primary cluster will be determined as non-noisy and nonchimeric.
The default value is </span>
                  <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">2</span>
                  <span class="ltx_text" style="font-size:90%;">.</span>
                </p>
</div>
</dd>
<dt id="Ch3.I2.ix2"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minpreplicate</span>
              </span></dt>
<dd>
<div id="Ch3.I2.ix2.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">Specified as decimal larger than 0.
If the proportion of replicates occured (number of replicates occured / total number of replicates of the same template) is equal to or larger than this value, such primary cluster will be determined as non-noisy and nonchimeric.
The default value is </span>
                  <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">1</span>
                  <span class="ltx_text" style="font-size:90%;">.</span>
                </p>
</div>
</dd>
<dt id="Ch3.I2.ix3"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minnpositive</span>
              </span></dt>
<dd>
<div id="Ch3.I2.ix3.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">Specified as integer larger than 0.
If the number of sequences determined as noisy or chimeric of a primary cluster is equal to or larger than this value, the cluster determined as noisy or chimeric in all the samples.
The default value is </span>
                  <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">1</span>
                  <span class="ltx_text" style="font-size:90%;">.</span>
                </p>
</div>
</dd>
<dt id="Ch3.I2.ix4"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minppositive</span>
              </span></dt>
<dd>
<div id="Ch3.I2.ix4.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">Specified as decimal equal to or larger than 0.
If the proportion of sequences determined as noisy or chimeric of a primary cluster (number of sequences determined as noisy or chimeric of a primary cluster / total number of sequences of a primary cluster) is equal to or larger than this value, the cluster determined as noisy or chimeric in all the samples.
The default value is </span>
                  <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">0</span>
                  <span class="ltx_text" style="font-size:90%;">.</span>
                </p>
</div>
</dd>
<dt id="Ch3.I2.ix5"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--runname</span>
              </span></dt>
<dd>
<div id="Ch3.I2.ix5.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">Specify RunID.
RunIDs in all sample names will be replaced to this RunID.
If multiple sample names become to the same, such samples will be integrated.</span>
                </p>
</div>
</dd>
</dl>
<p class="ltx_p">The <span class="ltx_text ltx_font_typewriter">--minnreplicate</span> and <span class="ltx_text ltx_font_typewriter">--minpreplicate</span> are arguments about intrasample judgement.
The <span class="ltx_text ltx_font_typewriter">--minnpositive</span> and <span class="ltx_text ltx_font_typewriter">--minppositive</span> are arguments about intersample judgement.
If both of <span class="ltx_text ltx_font_typewriter">--minnreplicate</span> and <span class="ltx_text ltx_font_typewriter">--minpreplicate</span> are fullfilled, the primary cluster will be determined as non-noisy and nonchimeric.
If both of <span class="ltx_text ltx_font_typewriter">--minnpositive</span> and <span class="ltx_text ltx_font_typewriter">--minppositive</span> are fullfilled, the primary cluster will be determined as noisy or chimeric.
The decision is common among all samples and different decision among samples are not allowed.
If there are samples that is not written in replicate list file, such sample is not used in this decision.</p>
</div>
<div id="Ch3.S1.SS1.p3" class="ltx_para">
<p class="ltx_p">If <span class="ltx_text ltx_font_typewriter">clcleanseqv</span> is executed like above, the following files will be saved additionally.</p>
<dl id="Ch3.I3" class="ltx_description">
<dt id="Ch3.I3.ix1"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text" style="font-size:90%;">primarycluster.chimeraremoved.fasta.gz</span>
              </span></dt>
<dd>
<div id="Ch3.I3.ix1.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">Representative sequences determined as nonchimeric.</span>
                </p>
</div>
</dd>
<dt id="Ch3.I3.ix2"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text" style="font-size:90%;">primarycluster.cleaned.fasta.gz</span>
              </span></dt>
<dd>
<div id="Ch3.I3.ix2.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">Representative sequences determined as non-noisy and nonchimeric</span>
                </p>
</div>
</dd>
<dt id="Ch3.I3.ix3"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text" style="font-size:90%;">RunID__TagID__PrimerID.chimericreads.txt.gz</span>
              </span></dt>
<dd>
<div id="Ch3.I3.ix3.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">List of sequences determined as chimeric</span>
                </p>
</div>
</dd>
</dl>
</div>
</section>
</section>
<section id="Ch3.S2" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">3.2 </span>Old pipeline using Assams</h3>

<div id="Ch3.S2.p1" class="ltx_para">
<p class="ltx_p">以下のコマンドでノイジー配列とキメラ配列の検出・除去が行えます。
なお、多数のファイルを一度に与えることができますが、できるだけ1ラン分のファイルを与えて下さい。
これは、ランごとに塩基配列決定の質にばらつきがあるためです1ラン分の配列が非常に多く(100万超)時間がかかってしまう場合は、1サンプルごとにこの処理を行って下さい。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clcleanseq \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">uchime --minh 0.1 --mindiv 0.8 end \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--detectmode=N+C \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--pnoisycluster=0.5 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile1 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">中略 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfileN \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
          </span>
</p>
<p class="ltx_p">オプションのうち、<span class="ltx_text ltx_font_typewriter">uchime</span>から<span class="ltx_text ltx_font_typewriter">end</span>の間はUCHIMEの実行オプションです。
<span class="ltx_text ltx_font_typewriter">--minh</span>と<span class="ltx_text ltx_font_typewriter">--mindiv</span>はキメラ配列と判定する感度に関するオプションです。
<span class="ltx_text ltx_font_typewriter">--minh</span>は値が小さいほど感度が高くなります。
UCHIMEの作者は0.1以上5以下の範囲にするよう推奨しています。
デフォルト値は0.1です。
<span class="ltx_text ltx_font_typewriter">--mindiv</span>の指定値が0.8の場合、0.8%以下の違いしかない配列間はキメラが形成されていても無視されます。
最終的に97%一致規準で配列をクラスタリングするとしたら、2%程度の違いしかない配列間のキメラはクラスタリング・コンセンサス配列作成によって潰されるため、この値を大きくすることで計算が速くなります。
また、<span class="ltx_text ltx_font_typewriter">--pnoisycluster</span>はノイジー配列と判定する感度に関するオプションです。
0以上1以下で指定して下さい。
大きいほど感度が上がります。
デフォルト値・推奨値は0.5です。
クラスタリングを最終的に97%以下の閾値で行うのであればこれでいいと思いますが、99%前後でクラスタリングする場合、<span class="ltx_text ltx_font_typewriter">--pnoisycluster</span>は0.9くらいにするといいかもしれません。
ただし、レアOTUを捨ててしまう可能性が高くなるので、1サンプル当たりの配列数を多めに読んでおく必要があります。</p>
</div>
<div id="Ch3.S2.p2" class="ltx_para">
<p class="ltx_p">各入力ファイルは同一の鋳型を同一のチューブ内でPCRした配列群を含んでいる必要があります。
Claidentがキメラ検出に利用しているUCHIMEでは、元配列数が多いコンセンサス配列ほどPCR当初から存在した＝キメラではないと仮定し、元配列数が少ないコンセンサス配列が、より元配列数の多いコンセンサス配列の部分配列の組み合わせで説明できるかどうかを検討します。
したがって、UCHIMEに「同一の鋳型を同一のチューブ内でPCRした配列群内でのコンセンサス配列の元配列数」を正確に伝える必要があるため、1ファイルは同一の鋳型を同一のチューブ内でPCRした配列群でなくてはならないのです。
これは1ラン分のファイルを一度に与えることよりもはるかに優先度が高いので、同一の鋳型を同一のチューブ内でPCRしたサンプルを複数回のランで塩基配列決定した場合などは複数ラン分のデータを入力しても構いません。</p>
</div>
<div id="Ch3.S2.p3" class="ltx_para">
<p class="ltx_p">UCHIMEによるキメラ判定はそれぞれのサンプルごとに行われ、複数のサンプルで観測されている完全一致配列の場合、どれか1つのサンプルでキメラと判定されていれば、他のサンプルでもキメラとみなして除去します。
ただし、下記のオプションでこの動作を変更することができます。
両オプションの両方を満たせばキメラと判定されます。</p>
<dl id="Ch3.I4" class="ltx_description">
<dt id="Ch3.I4.ix1"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minnpositive</span>
            </span></dt>
<dd>
<div id="Ch3.I4.ix1.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">整数値で指定。
この値以上の配列(サンプルではない)がキメラと判定されていれば、全サンプルでキメラと見なす。
デフォルト値は</span>
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">1</span>
                <span class="ltx_text" style="font-size:90%;">。</span>
              </p>
</div>
</dd>
<dt id="Ch3.I4.ix2"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minppositive</span>
            </span></dt>
<dd>
<div id="Ch3.I4.ix2.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">小数値で指定。
キメラと判定された配列数/完全一致した全配列数がこの値以上であれば、全サンプルでキメラと見なす。
デフォルト値は</span>
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">0</span>
                <span class="ltx_text" style="font-size:90%;">。</span>
              </p>
</div>
</dd>
</dl>
<p class="ltx_p">出力フォルダには、以下のファイル群が保存されます。</p>
<dl id="Ch3.I5" class="ltx_description">
<dt id="Ch3.I5.ix1"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">RunID__TagID__PrimerID.chimeraremoved.dereplicated.fastq.gz</span>
            </span></dt>
<dd>
<div id="Ch3.I5.ix1.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">キメラ配列除去し完全一致配列をまとめた配列</span>
              </p>
</div>
</dd>
<dt id="Ch3.I5.ix2"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">RunID__TagID__PrimerID.chimeraremoved.fastq.gz</span>
            </span></dt>
<dd>
<div id="Ch3.I5.ix2.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">キメラ配列除去した配列</span>
              </p>
</div>
</dd>
<dt id="Ch3.I5.ix3"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">RunID__TagID__PrimerID.denoised.dereplicated.fastq.gz</span>
            </span></dt>
<dd>
<div id="Ch3.I5.ix3.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">ノイジー配列除去し完全一致配列をまとめた配列</span>
              </p>
</div>
</dd>
<dt id="Ch3.I5.ix4"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">RunID__TagID__PrimerID.denoised.fastq.gz</span>
            </span></dt>
<dd>
<div id="Ch3.I5.ix4.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">ノイジー配列除去した配列</span>
              </p>
</div>
</dd>
<dt id="Ch3.I5.ix5"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">RunID__TagID__PrimerID.cleaned.dereplicated.fastq.gz</span>
            </span></dt>
<dd>
<div id="Ch3.I5.ix5.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">キメラ配列とノイジー配列を除去し完全一致配列をまとめた配列</span>
              </p>
</div>
</dd>
<dt id="Ch3.I5.ix6"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">RunID__TagID__PrimerID.cleaned.fastq.gz</span>
            </span></dt>
<dd>
<div id="Ch3.I5.ix6.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">キメラ配列とノイジー配列を除去した配列</span>
              </p>
</div>
</dd>
<dt id="Ch3.I5.ix7"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">RunID__TagID__PrimerID.chimericreads.txt.gz</span>
            </span></dt>
<dd>
<div id="Ch3.I5.ix7.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">キメラ配列と判定された配列のリスト</span>
              </p>
</div>
</dd>
<dt id="Ch3.I5.ix8"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">RunID__TagID__PrimerID.noisyreads.txt.gz</span>
            </span></dt>
<dd>
<div id="Ch3.I5.ix8.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">ノイジー配列と判定された配列のリスト</span>
              </p>
</div>
</dd>
<dt id="Ch3.I5.ix9"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">RunID__TagID__PrimerID.singletons.txt.gz</span>
            </span></dt>
<dd>
<div id="Ch3.I5.ix9.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">完全一致配列をまとめたときにシングルトンになった配列のリスト</span>
              </p>
</div>
</dd>
<dt id="Ch3.I5.ix10"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">RunID__TagID__PrimerID.uchime.txt.gz</span>
            </span></dt>
<dd>
<div id="Ch3.I5.ix10.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">UCHIMEによるキメラ判定結果のログ</span>
              </p>
</div>
</dd>
<dt id="Ch3.I5.ix11"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">RunID__TagID__PrimerID.parameter.txt</span>
            </span></dt>
<dd>
<div id="Ch3.I5.ix11.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">まとまった完全一致配列をノイジーと判定するのに使用した所属配列数の下限値</span>
              </p>
</div>
</dd>
</dl>
<p class="ltx_p">他にも多くのファイルが出力されますが、後の解析に必要なものもありますので削除しないようにして下さい。
キメラ配列除去を無効化してノイジー配列除去のみを行う場合は<span class="ltx_text ltx_font_typewriter">--detectmode=noise</span>、逆にノイジー配列除去を無効化してキメラ配列除去のみを行う場合は<span class="ltx_text ltx_font_typewriter">--detectmode=chimera</span>オプションを付けて実行して下さい。
ただし、<span class="ltx_text ltx_font_typewriter">--detectmode=noise</span>で処理した結果を<span class="ltx_text ltx_font_typewriter">--detectmode=chimera</span>で処理しても、一度にノイジー配列・キメラ配列除去を行った場合(<span class="ltx_text ltx_font_typewriter">--detectmode=N+C</span>を指定した場合またはデフォルト設定)と結果は一致しませんのでご注意下さい。
処理する場合は同時に行う必要があります。
これは、キメラ配列もノイジー配列と判定するための情報を持っており、同時処理の場合にはそれを利用するためです。</p>
</div>
<section id="Ch3.S2.SS1" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">3.2.1 </span>Noisy and/or chimeric sequence removal based on PCR replicates</h4>

<div id="Ch3.S2.SS1.p1" class="ltx_para">
<p class="ltx_p">PCRレプリケートを用意してノイジー配列・キメラ配列を検出・除去するには、まず同一鋳型DNA由来のPCRレプリケートはどのサンプルとどのサンプルなのかを記したファイルを用意する必要があります。
以下のような内容で用意して下さい。</p>
<p class="ltx_p ltx_align_left" style="background-color:#E6E6E6;padding-top:12pt;padding-bottom:12pt;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| sample1  sample2  sample3</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| sample4  sample5</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| sample6  sample7</span>
            </span>
</p>
<p class="ltx_p">同一の行内にタブ区切りで書かれたサンプルが、同一鋳型由来のPCRレプリケートであることを表しています。
異なる鋳型由来のサンプルは別の行に記述して下さい。
PCRレプリケートは3つ以上あっても構いません。
また、レプリケート数が鋳型ごとに異なっても構いません。</p>
</div>
<div id="Ch3.S2.SS1.p2" class="ltx_para">
<p class="ltx_p">上記ファイルが用意できたら、以下のように<span class="ltx_text ltx_font_typewriter">clcleanseq</span>を実行することでレプリケート間で共通しないプライマリクラスタをキメラもしくはノイジーであると判断して除去します。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clcleanseq \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--replicatelist=PCRレプリケート指示ファイル \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">uchime --minh 0.1 --mindiv 0.8 end \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--detectmode=N+C \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--pnoisycluster=0.5 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile1 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">中略 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfileN \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
            </span>
</p>
<p class="ltx_p">レプリケートが3つ以上であっても、全てのレプリケートで検出されたプライマリクラスタのみがノイジーでもキメラでもないと判断されます。
また、1つのプライマリクラスタが複数の鋳型から検出されていた場合、どれか1つの鋳型でノイジーまたはキメラと判定されていれば、他の鋳型でもノイジーまたはキメラとみなして除去します。
これらの設定は以下のオプションで変更することができます。</p>
<dl id="Ch3.I6" class="ltx_description">
<dt id="Ch3.I6.ix1"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minnreplicate</span>
              </span></dt>
<dd>
<div id="Ch3.I6.ix1.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">整数値で指定。
この値以上のレプリケートで観測されたプライマリクラスタはそのサンプルではノイジーでもキメラでもないと見なす。
デフォルト値は</span>
                  <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">2</span>
                  <span class="ltx_text" style="font-size:90%;">。</span>
                </p>
</div>
</dd>
<dt id="Ch3.I6.ix2"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minpreplicate</span>
              </span></dt>
<dd>
<div id="Ch3.I6.ix2.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">小数値で指定。
プライマリクラスタが観測されたレプリケート数/サンプルの総レプリケート数がこの値以上であれば、そのサンプルではノイジーでもキメラでもないと見なす。
デフォルト値は</span>
                  <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">1</span>
                  <span class="ltx_text" style="font-size:90%;">。</span>
                </p>
</div>
</dd>
<dt id="Ch3.I6.ix3"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minnpositive</span>
              </span></dt>
<dd>
<div id="Ch3.I6.ix3.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">整数値で指定。
この値以上の配列(サンプルではない)がノイジーまたはキメラと判定されていれば、全サンプルでノイジーまたはキメラと見なす。
デフォルト値は</span>
                  <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">1</span>
                  <span class="ltx_text" style="font-size:90%;">。</span>
                </p>
</div>
</dd>
<dt id="Ch3.I6.ix4"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minppositive</span>
              </span></dt>
<dd>
<div id="Ch3.I6.ix4.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">小数値で指定。
ノイジーまたはキメラと判定された配列数/プライマリクラスタの総配列数がこの値以上であれば、全サンプルでノイジーまたはキメラと見なす。
デフォルト値は</span>
                  <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">0</span>
                  <span class="ltx_text" style="font-size:90%;">。</span>
                </p>
</div>
</dd>
<dt id="Ch3.I6.ix5"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--runname</span>
              </span></dt>
<dd>
<div id="Ch3.I6.ix5.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">RunIDを指定。
サンプル名に含まれるRunIDが全てこれに置換される。
同一サンプル名になったサンプルは統合される。</span>
                </p>
</div>
</dd>
</dl>
<p class="ltx_p"><span class="ltx_text ltx_font_typewriter">--minnreplicate</span>と<span class="ltx_text ltx_font_typewriter">--minpreplicate</span>の値に基づく1サンプル内での判定と、<span class="ltx_text ltx_font_typewriter">--minnpositive</span>と<span class="ltx_text ltx_font_typewriter">--minppositive</span>の値に基づく全サンプルにまたがる判定が2段階で行われていることに注意して下さい。
前者では2つの条件を両方満たすとノイジーでもキメラでもないと判定され、後者では2つの条件を両方満たすとノイジーまたはキメラであると判定されます。
また、同一のプライマリクラスタが、あるサンプルではノイジーまたはキメラだが、別のサンプルではそうでない、といった判定は行えません。
最終的な判定は全サンプル共通です。
また、PCRレプリケート指示ファイルに記されていないサンプルでは判定が行われません。</p>
</div>
</section>
</section>
</section>
<section id="Ch4" class="ltx_chapter">
<h2 class="ltx_title ltx_title_chapter">
<span class="ltx_tag ltx_tag_chapter">Chapter 4 </span>OTU picking based on nucleotide sequence clustering</h2>

<section id="Ch4.S1" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">4.1 </span>Intra-sample clustering</h3>

<section id="Ch4.S1.SS1" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">4.1.1 </span>New pipeline using VSEARCH</h4>

<div id="Ch4.S1.SS1.p1" class="ltx_para">
<p class="ltx_p">新パイプラインではこのステップはありません。</p>
</div>
</section>
<section id="Ch4.S1.SS2" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">4.1.2 </span>Old pipeline using Assams</h4>

<div id="Ch4.S1.SS2.p1" class="ltx_para">
<p class="ltx_p">以下のコマンドで、同一の鋳型を同一のチューブ内でPCRした配列群のクラスタリングを行います。
前節の結果得られる、完全一致配列をまとめた結果(<span class="ltx_text ltx_font_typewriter">RunID__TagID__PrimerID.cleaned.dereplicated.fastq.gz</span>)を入力します。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clclassseq \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minident=0 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--strand=plus \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile1 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">中略 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfileN \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
            </span>
</p>
<p class="ltx_p">前節の出力ファイルに対してこの処理を適用する場合、以下のようにコマンドを打てばいいでしょう。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clclassseq \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minident=0 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--strand=plus \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">"前節の出力フォルダ/*.cleaned.dereplicated.fastq.gz" \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
            </span>
</p>
<p class="ltx_p">オプションのうち、<span class="ltx_text ltx_font_typewriter">--minident=0</span>は不一致が1塩基以下の配列をまとめる設定です(実際には、類似度が(最短配列の長さ-11/最短配列の長さ-10)以上の配列をまとめます。
最長配列の長さが400bpで最短配列の長さが350bpなら、0.997となり、400bpの配列でも1塩基しか不一致は許容されませんので意図通りになりますが、最短配列と最長配列の差が大きすぎると意図した通りになりません。
そのため、最短配列は最長配列の半分より11塩基以上長くなくてはなりません)。
最終的に97%で配列をまとめる場合もこの時点では0にしておくのがおすすめです。
最終的に99%でまとめる場合にもここでは0にしておきます。
<span class="ltx_text ltx_font_typewriter">--strand=plus</span>は同一ストランドの比較だけでクラスタリングを行うオプションです。
片側からしか読んでいないので、両ストランドの比較をする必要はなく、比較するとかえって異常なクラスタリングをしかねません。
両側から読んでいる場合も、連結した後はストランドは揃っているのでこれでいいはずです。
出力フォルダ内の<span class="ltx_text ltx_font_typewriter">*.assembled.fastq.gz</span>がクラスタリング後の配列です。</p>
</div>
</section>
</section>
<section id="Ch4.S2" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">4.2 </span>Inter-sample clustering</h3>

<section id="Ch4.S2.SS1" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">4.2.1 </span>New pipeline using VSEARCH</h4>

<div id="Ch4.S2.SS1.p1" class="ltx_para">
<p class="ltx_p">塩基配列のクラスタリングを行ってOTUを作成するには、以下のコマンドを実行して下さい。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clclassseqv \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minident=0.97 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile1 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">中略 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfileN \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
            </span>
</p>
<p class="ltx_p">入力ファイルには、<span class="ltx_text ltx_font_typewriter">clcleanseqv</span>の実行時にPCRレプリケート法によるノイジー配列・キメラ配列の除去を併用した場合は<span class="ltx_text ltx_font_typewriter">primarycluster.cleaned.fasta.gz</span>を、併用しなかった場合には<span class="ltx_text ltx_font_typewriter">primarycluster.denoised.fasta.gz</span>を与えて下さい。
オーバーラップのないペアエンドデータを<span class="ltx_text ltx_font_typewriter">clconcatpair</span>を用いて連結した配列では、<span class="ltx_text ltx_font_typewriter">--paddinglen=16</span>をオプションとして加えて実行して下さい。</p>
</div>
<div id="Ch4.S2.SS1.p2" class="ltx_para">
<p class="ltx_p">出力フォルダには、以下のファイルが保存されています。</p>
<dl id="Ch4.I1" class="ltx_description">
<dt id="Ch4.I1.ix1"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text" style="font-size:90%;">clustered.otu.gz</span>
              </span></dt>
<dd>
<div id="Ch4.I1.ix1.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">どの生配列がどのOTUに割り当てられたかを記録しているファイル</span>
                </p>
</div>
</dd>
<dt id="Ch4.I1.ix2"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text" style="font-size:90%;">clustered.fasta.gz</span>
              </span></dt>
<dd>
<div id="Ch4.I1.ix2.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">OTUの代表配列</span>
                </p>
</div>
</dd>
</dl>
</div>
</section>
<section id="Ch4.S2.SS2" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">4.2.2 </span>Old pipeline using Assams</h4>

<div id="Ch4.S2.SS2.p1" class="ltx_para">
<p class="ltx_p">サンプル内クラスタリングデータを用いてサンプル間でのクラスタリングを行い、全体のクラスタリング結果を得るには以下のコマンドを実行します。
1サンプルしかない場合も、上記のサンプル内配列クラスタリングを行ってからこの処理を行って下さい。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clclassclass \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minident=0 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--strand=plus \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile1 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">中略 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfileN \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder1↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clreclassclass \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minident=0.99 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--strand=plus \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder1 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder2↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clreclassclass \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minident=0.97 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--strand=plus \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder2 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder3↓</span>
            </span>
</p>
<p class="ltx_p">ここでの入力ファイルは前節の出力ファイル<span class="ltx_text ltx_font_typewriter">*.assembled.fastq.gz</span>です。
したがって、実際には以下のように入力することになるでしょう。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clclassclass \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minident=0 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--strand=plus \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">"前節の出力フォルダ/*.assembled.fastq.gz" \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder1↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clreclassclass \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minident=0.99 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--strand=plus \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder1 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder2↓</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clreclassclass \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minident=0.97 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--strand=plus \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder2 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder3↓</span>
            </span>
</p>
<p class="ltx_p">1つ目の<span class="ltx_text ltx_font_typewriter">clclassclass</span>でサンプル内配列クラスタリングでまとめられた配列において、不一致が1塩基以下の配列をまとめます。
2つ目の<span class="ltx_text ltx_font_typewriter">clreclassclass</span>で、99%以上一致する配列をまとめ直します。
さらに3つ目の<span class="ltx_text ltx_font_typewriter">clreclassclass</span>で、97%以上一致する配列をまとめ直します。
こうすることで、より高速に、より正確にクラスタリングが行われます。
<span class="ltx_text ltx_font_typewriter">clreclassclass</span>を使用せずに<span class="ltx_text ltx_font_typewriter">clclassclass</span>に97%でまとめるよう指示してしまうと、アルゴリズム上「まとめすぎ」が発生しやすくなります。</p>
</div>
<div id="Ch4.S2.SS2.p2" class="ltx_para">
<p class="ltx_p">出力フォルダには、以下のファイルが保存されています。</p>
<dl id="Ch4.I2" class="ltx_description">
<dt id="Ch4.I2.ix1"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text" style="font-size:90%;">assembled.contigmembers.gz</span>
              </span></dt>
<dd>
<div id="Ch4.I2.ix1.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">どの生配列がどのOTUに割り当てられたかを記録しているファイル</span>
                </p>
</div>
</dd>
<dt id="Ch4.I2.ix2"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text" style="font-size:90%;">assembled.fastq.gz</span>
              </span></dt>
<dd>
<div id="Ch4.I2.ix2.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">OTUのコンセンサス配列</span>
                </p>
</div>
</dd>
<dt id="Ch4.I2.ix3"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text" style="font-size:90%;">assembled.fasta</span>
              </span></dt>
<dd>
<div id="Ch4.I2.ix3.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">OTUのコンセンサス配列</span>
                </p>
</div>
</dd>
</dl>
</div>
</section>
</section>
<section id="Ch4.S3" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">4.3 </span>Mapping raw sequencing reads to representative sequences of the clusters</h3>

<div id="Ch4.S3.p1" class="ltx_para">
<p class="ltx_p">ノイジー配列・キメラ配列と判定されてしまった配列でも、OTUの代表配列に対して指定した類似度の閾値で貼り付けることが可能なのであれば、復活させることができます。
これにより、データの減少を抑えることができます。</p>
</div>
<section id="Ch4.S3.SS1" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">4.3.1 </span>New pipeline using VSEARCH</h4>

<div id="Ch4.S3.SS1.p1" class="ltx_para">
<p class="ltx_p">以下のコマンドを実行して下さい。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clrecoverseqv \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minident=0.97 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--centroid=OTU代表配列のファイル \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile1 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">中略 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfileN \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
            </span>
</p>
<p class="ltx_p">OTU代表配列のファイルには<span class="ltx_text ltx_font_typewriter">clclassseqv</span>の出力した<span class="ltx_text ltx_font_typewriter">clustered.fasta.gz</span>を用います。
入力ファイルとしては、<span class="ltx_text ltx_font_typewriter">clcleanseqv</span>の出力した<span class="ltx_text ltx_font_typewriter">primarycluster.fasta.gz</span>または<span class="ltx_text ltx_font_typewriter">clcleanseqv</span>での処理や低品質配列の除去を行う前の配列ファイルを使用して下さい。
出力フォルダには、<span class="ltx_text ltx_font_typewriter">clclassseqv</span>と同様のファイル群が作成されます。
なお、オーバーラップのないペアエンドデータを<span class="ltx_text ltx_font_typewriter">clconcatpair</span>を用いて連結した配列では、<span class="ltx_text ltx_font_typewriter">--paddinglen=16</span>をオプションとして加えて実行して下さい。</p>
</div>
</section>
<section id="Ch4.S3.SS2" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">4.3.2 </span>Old pipeline using Assams</h4>

<div id="Ch4.S3.SS2.p1" class="ltx_para">
<p class="ltx_p">旧パイプラインではこのステップはありません。</p>
</div>
</section>
</section>
</section>
<section id="Ch5" class="ltx_chapter">
<h2 class="ltx_title ltx_title_chapter">
<span class="ltx_tag ltx_tag_chapter">Chapter 5 </span>Summarizing and post-processing of OTU picking results</h2>

<section id="Ch5.S1" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">5.1 </span>Making summary table</h3>

<div id="Ch5.S1.p1" class="ltx_para">
<p class="ltx_p">以下のコマンドで、クラスタリング結果から各サンプルにおける各OTUに割り当てられた生配列数を表にすることができます。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clsumclass \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--output=Matrix \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile↓</span>
          </span>
</p>
<p class="ltx_p">ここでの入力ファイルは、クラスタリング結果の出力フォルダに保存されている、<span class="ltx_text ltx_font_typewriter">clustered.otu.gz</span> (新パイプライン)または<span class="ltx_text ltx_font_typewriter">assembled.contigmembers.gz</span> (旧パイプライン)というファイルです。
出力されるファイルは表<a href="#Ch5.T1" title="Table  5.1 ‣ 5.1 Making summary table ‣ Chapter 5 Summarizing and post-processing of OTU picking results ‣ Matabarcoding and DNA barcoding for Ecologists" class="ltx_ref"><span class="ltx_text ltx_ref_tag">5.1</span></a>のようなタブ区切りのテキストファイルで、Excelなどの表計算ソフトやRで読み込むことができます。
ただし、表計算ソフトは巨大な行列の全てを読み込むことができない可能性がありますので注意して下さい。
このファイルに基づいて、群集生態学的な解析を行うことができます。</p>
</div>
<figure id="Ch5.T1" class="ltx_table">
<table class="ltx_tabular ltx_centering ltx_align_middle">
<tbody class="ltx_tbody">
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">samplename</span>
              </th>
<td class="ltx_td ltx_align_right">
                <span class="ltx_text" style="font-size:80%;">amongsampleotu_1</span>
              </td>
<td class="ltx_td ltx_align_right">
                <span class="ltx_text" style="font-size:80%;">amongsampleotu_2</span>
              </td>
<td class="ltx_td ltx_align_right">
                <span class="ltx_text" style="font-size:80%;">amongsampleotu_3</span>
              </td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r ltx_border_tt">
                <span class="ltx_text" style="font-size:80%;">sampleA</span>
              </th>
<td class="ltx_td ltx_align_right ltx_border_tt">
                <span class="ltx_text" style="font-size:80%;">2371</span>
              </td>
<td class="ltx_td ltx_align_right ltx_border_tt">
                <span class="ltx_text" style="font-size:80%;">0</span>
              </td>
<td class="ltx_td ltx_align_right ltx_border_tt">
                <span class="ltx_text" style="font-size:80%;">0</span>
              </td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">sampleB</span>
              </th>
<td class="ltx_td ltx_align_right">
                <span class="ltx_text" style="font-size:80%;">0</span>
              </td>
<td class="ltx_td ltx_align_right">
                <span class="ltx_text" style="font-size:80%;">1518</span>
              </td>
<td class="ltx_td ltx_align_right">
                <span class="ltx_text" style="font-size:80%;">0</span>
              </td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">sampleC</span>
              </th>
<td class="ltx_td ltx_align_right">
                <span class="ltx_text" style="font-size:80%;">1398</span>
              </td>
<td class="ltx_td ltx_align_right">
                <span class="ltx_text" style="font-size:80%;">0</span>
              </td>
<td class="ltx_td ltx_align_right">
                <span class="ltx_text" style="font-size:80%;">0</span>
              </td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">sampleD</span>
              </th>
<td class="ltx_td ltx_align_right">
                <span class="ltx_text" style="font-size:80%;">0</span>
              </td>
<td class="ltx_td ltx_align_right">
                <span class="ltx_text" style="font-size:80%;">1436</span>
              </td>
<td class="ltx_td ltx_align_right">
                <span class="ltx_text" style="font-size:80%;">0</span>
              </td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">sampleE</span>
              </th>
<td class="ltx_td ltx_align_right">
                <span class="ltx_text" style="font-size:80%;">0</span>
              </td>
<td class="ltx_td ltx_align_right">
                <span class="ltx_text" style="font-size:80%;">0</span>
              </td>
<td class="ltx_td ltx_align_right">
                <span class="ltx_text" style="font-size:80%;">1360</span>
              </td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">sampleF</span>
              </th>
<td class="ltx_td ltx_align_right">
                <span class="ltx_text" style="font-size:80%;">0</span>
              </td>
<td class="ltx_td ltx_align_right">
                <span class="ltx_text" style="font-size:80%;">0</span>
              </td>
<td class="ltx_td ltx_align_right">
                <span class="ltx_text" style="font-size:80%;">977</span>
              </td>
</tr>
</tbody>
</table>
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_table">Table  5.1: </span>集計表の例 — 数値は各サンプルにおける各OTUの観測配列数です。</figcaption>
</figure>
</section>
<section id="Ch5.S2" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">5.2 </span>Excluding specified OTUs and/or samples from summary table</h3>

<div id="Ch5.S2.p1" class="ltx_para">
<p class="ltx_p"><span class="ltx_text ltx_font_typewriter">clsumclass</span>では、1回だけ出現した配列も含めて全て出力されますが、以下のコマンドを用いてサンプルやOTUを選別することができます。
入力ファイルは<span class="ltx_text ltx_font_typewriter">clsumclass</span>の出力ファイルです。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clfiltersum \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">オプション \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile↓</span>
          </span>
</p>
<p class="ltx_p">使用できるオプションは以下の通りです。</p>
<dl id="Ch5.I1" class="ltx_description">
<dt id="Ch5.I1.ix1"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minnseqotu</span>
            </span></dt>
<dd>
<div id="Ch5.I1.ix1.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">整数値で指定。
この値に基づいて「割り当てられた生配列数がどのサンプルでも指定値未満のOTU」を除去します。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I1.ix2"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minpseqotu</span>
            </span></dt>
<dd>
<div id="Ch5.I1.ix2.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">小数値で指定。
この値に基づいて「割り当てられた生配列数/サンプルの生配列総数がどのサンプルでも指定値未満のOTU」を除去します。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I1.ix3"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minntotalseqotu</span>
            </span></dt>
<dd>
<div id="Ch5.I1.ix3.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">整数値で指定。
この値に基づいて「割り当てられた生配列総数が指定値未満のOTU」を除去します。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I1.ix4"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minnseqsample</span>
            </span></dt>
<dd>
<div id="Ch5.I1.ix4.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">整数値で指定。
この値に基づいて「どのOTUの生配列数も指定値未満のサンプル」を除去します。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I1.ix5"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minpseqsample</span>
            </span></dt>
<dd>
<div id="Ch5.I1.ix5.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">小数値で指定。
この値に基づいて「そのサンプルのそのOTUに割り当てられた生配列数/OTUの生配列総数がどのOTUでも指定値未満のサンプル」を除去します。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I1.ix6"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minntotalseqsample</span>
            </span></dt>
<dd>
<div id="Ch5.I1.ix6.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">整数値で指定。
この値に基づいて「生配列総数が指定値未満のサンプル」を除去します。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I1.ix7"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--otu</span>
            </span></dt>
<dd>
<div id="Ch5.I1.ix7.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">残したいOTU名をカンマで区切って指定します。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I1.ix8"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--negativeotu</span>
            </span></dt>
<dd>
<div id="Ch5.I1.ix8.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">除去したいOTU名をカンマで区切って指定します。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I1.ix9"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--otulist</span>
            </span></dt>
<dd>
<div id="Ch5.I1.ix9.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">ファイル名を指定。
1行に1つのOTU名を記したテキストファイルとして残したいOTU名を指定します。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I1.ix10"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--negativeotulist</span>
            </span></dt>
<dd>
<div id="Ch5.I1.ix10.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">ファイル名を指定。
1行に1つのOTU名を記したテキストファイルとして除去したいOTU名を指定します。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I1.ix11"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--otuseq</span>
            </span></dt>
<dd>
<div id="Ch5.I1.ix11.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">ファイル名を指定。
FASTA形式の配列ファイルとして残したいOTU名を指定します。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I1.ix12"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--negativeotuseq</span>
            </span></dt>
<dd>
<div id="Ch5.I1.ix12.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">ファイル名を指定。
FASTA形式の配列ファイルとして除去したいOTU名を指定します。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I1.ix13"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--sample</span>
            </span></dt>
<dd>
<div id="Ch5.I1.ix13.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">残したいサンプル名をカンマで区切って指定します。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I1.ix14"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--negativesample</span>
            </span></dt>
<dd>
<div id="Ch5.I1.ix14.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">除去したいサンプル名をカンマで区切って指定します。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I1.ix15"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--samplelist</span>
            </span></dt>
<dd>
<div id="Ch5.I1.ix15.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">ファイル名を指定。
1行に1つのサンプル名を記したテキストファイルとして残したいサンプル名を指定します。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I1.ix16"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--negativesamplelist</span>
            </span></dt>
<dd>
<div id="Ch5.I1.ix16.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">ファイル名を指定。
1行に1つのサンプル名を記したテキストファイルとして除去したいサンプル名を指定します。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I1.ix17"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--replicatelist</span>
            </span></dt>
<dd>
<div id="Ch5.I1.ix17.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">ファイル名を指定。
PCRレプリケート関係にあるサンプルをタブ区切りで同一行に記述する。
出力で1つのサンプルに統合される。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I1.ix18"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--runname</span>
            </span></dt>
<dd>
<div id="Ch5.I1.ix18.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">RunIDを指定。
集計表中のサンプル名に含まれるRunIDが全てこれに置換される。
同一サンプル名になったサンプルは統合される。</span>
              </p>
</div>
</dd>
</dl>
</div>
<div id="Ch5.S2.p2" class="ltx_para">
<p class="ltx_p">後述するノイジー配列・キメラ配列の除去や他領域配列の除去を行った上で、残った配列を用いて<span class="ltx_text ltx_font_typewriter">--otuseq</span>などのオプションを用いたOTUの選別を行う場合は、他の選別より先に行って下さい。</p>
</div>
</section>
<section id="Ch5.S3" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">5.3 </span>Noisy and/or chimeric sequence removal based on PCR replicates after OTU picking</h3>

<div id="Ch5.S3.p1" class="ltx_para">
<p class="ltx_p">新パイプラインでは<span class="ltx_text ltx_font_typewriter">clcleanseqv</span>の実行時に行っていますが、旧パイプラインでもPCRレプリケートを用意してノイジー配列・キメラ配列を検出・除去することが可能です。
それには、まず同一鋳型DNA由来のPCRレプリケートはどのサンプルとどのサンプルなのかを記したファイルを用意する必要があります。
以下のような内容で用意して下さい。</p>
<p class="ltx_p ltx_align_left" style="background-color:#E6E6E6;padding-top:12pt;padding-bottom:12pt;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| sample1  sample2  sample3</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| sample4  sample5</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">| sample6  sample7</span>
          </span>
</p>
<p class="ltx_p">同一の行内にタブ区切りで書かれたサンプルが、同一鋳型由来のPCRレプリケートであることを表しています。
異なる鋳型由来のサンプルは別の行に記述して下さい。
PCRレプリケートは3つ以上あっても構いません。
また、レプリケート数が鋳型ごとに異なっても構いません。</p>
</div>
<div id="Ch5.S3.p2" class="ltx_para">
<p class="ltx_p">上記ファイルが用意できたら、以下のコマンドでクラスタリング結果からレプリケート間で共通しない配列を除去します。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clfilterseq \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--contigmembers=*.contigmembers.gz \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--replicatelist=PCRレプリケート指示ファイル \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile↓</span>
          </span>
</p>
<p class="ltx_p">レプリケートが3つ以上であっても、全てのレプリケートで検出された配列のみがノイジーでもキメラでもないと判断されます。
また、1つのOTUが複数の鋳型から検出されていた場合、どれか1つの鋳型でノイジーまたはキメラと判定されていれば、他の鋳型でもノイジーまたはキメラとみなして除去します。
これらの設定は以下のオプションで変更することができます。</p>
<dl id="Ch5.I2" class="ltx_description">
<dt id="Ch5.I2.ix1"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minnreplicate</span>
            </span></dt>
<dd>
<div id="Ch5.I2.ix1.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">整数値で指定。
この値以上のレプリケートで観測されたOTUはそのサンプルではノイジーでもキメラでもないと見なす。
デフォルト値は</span>
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">2</span>
                <span class="ltx_text" style="font-size:90%;">。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I2.ix2"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minpreplicate</span>
            </span></dt>
<dd>
<div id="Ch5.I2.ix2.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">小数値で指定。
OTUが観測されたレプリケート数/サンプルの総レプリケート数がこの値以上であれば、そのサンプルではノイジーでもキメラでもないと見なす。
デフォルト値は</span>
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">1</span>
                <span class="ltx_text" style="font-size:90%;">。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I2.ix3"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minnpositive</span>
            </span></dt>
<dd>
<div id="Ch5.I2.ix3.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">整数値で指定。
この値以上の配列(サンプルではない)がノイジーまたはキメラと判定されていれば、全サンプルでノイジーまたはキメラと見なす。
デフォルト値は</span>
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">1</span>
                <span class="ltx_text" style="font-size:90%;">。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I2.ix4"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--minppositive</span>
            </span></dt>
<dd>
<div id="Ch5.I2.ix4.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">小数値で指定。
ノイジーまたはキメラと判定された配列数/OTUの総配列数がこの値以上であれば、全サンプルでノイジーまたはキメラと見なす。
デフォルト値は</span>
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">0</span>
                <span class="ltx_text" style="font-size:90%;">。</span>
              </p>
</div>
</dd>
<dt id="Ch5.I2.ix5"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">--runname</span>
            </span></dt>
<dd>
<div id="Ch5.I2.ix5.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">RunIDを指定。
サンプル名に含まれるRunIDが全てこれに置換される。
同一サンプル名になったサンプルは統合される。</span>
              </p>
</div>
</dd>
</dl>
<p class="ltx_p"><span class="ltx_text ltx_font_typewriter">--minnreplicate</span>と<span class="ltx_text ltx_font_typewriter">--minpreplicate</span>の値に基づく1サンプル内での判定と、<span class="ltx_text ltx_font_typewriter">--minnpositive</span>と<span class="ltx_text ltx_font_typewriter">--minppositive</span>の値に基づく全サンプルにまたがる判定が2段階で行われていることに注意して下さい。
前者では2つの条件を両方満たすとノイジーでもキメラでもないと判定され、後者では2つの条件を両方満たすとノイジーまたはキメラであると判定されます。
また、同一のOTUが、あるサンプルではノイジーまたはキメラだが、別のサンプルではそうでない、といった判定は行えません。
最終的な判定は全サンプル共通です。
また、PCRレプリケート指示ファイルに記されていないサンプルでは判定が行われません。</p>
</div>
<div id="Ch5.S3.p3" class="ltx_para">
<p class="ltx_p">ここで作成したFASTA配列ファイルを第<a href="#Ch5.S2" title="5.2 Excluding specified OTUs and/or samples from summary table ‣ Chapter 5 Summarizing and post-processing of OTU picking results ‣ Matabarcoding and DNA barcoding for Ecologists" class="ltx_ref"><span class="ltx_text ltx_ref_tag">5.2</span></a>節の<span class="ltx_text ltx_font_typewriter">clfiltersum</span>の<span class="ltx_text ltx_font_typewriter">--otuseq</span>オプションに指定して集計表を加工することで、この配列ファイルから除去されたOTUを集計表からも除去することができます。
また、<span class="ltx_text ltx_font_typewriter">--replicatelist</span>オプションと<span class="ltx_text ltx_font_typewriter">--runname</span>オプションを<span class="ltx_text ltx_font_typewriter">clfilterseq</span>と同様に指定すれば、PCRレプリケート関係にあるサンプルは1つのサンプルに統合されます(セルの値は合算されます)。</p>
</div>
</section>
<section id="Ch5.S4" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">5.4 </span>Chimera removal using UCHIME algorithm</h3>

<div id="Ch5.S4.p1" class="ltx_para">
<p class="ltx_p">旧パイプラインではクラスタリング前にもUCHIMEでキメラ配列除去は行っていますが、クラスタリング後の配列に対してリファレンスなしでのキメラ配列除去とリファレンスありでのキメラ配列除去を追加で行うことができます。
両方行う場合は、リファレンスなしでのキメラ配列除去を先に行ってから、リファレンスありでのキメラ配列除去を行います。
両方のキメラ配列除去を行う場合は、クラスタリング前のノイジー配列・キメラ配列除去において、キメラ配列除去を無効にしてもいいかもしれません。
クラスタリング後のキメラ配列除去の方がずっと高速なので、1サンプル当たりの配列が膨大でクラスタリング前のキメラ配列除去では長時間を要する場合には、この方法の方が適しています。
なお、PCRレプリケート法によるノイジー配列・キメラ配列の除去も併用する場合は、PCRレプリケート法によるノイジー配列・キメラ配列の除去を最初に行います。
利用するコマンドは<span class="ltx_text ltx_font_typewriter">clrunuchime</span>です。
このコマンドでは、UCHIMEアルゴリズムを実装しているVSEARCHというプログラムを呼び出して、キメラ配列除去を行います。</p>
</div>
<section id="Ch5.S4.SS1" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">5.4.1 </span><span class="ltx_text ltx_font_italic">De novo</span> chimera removal</h4>

<div id="Ch5.S4.SS1.p1" class="ltx_para">
<p class="ltx_p"><span class="ltx_text ltx_font_typewriter">clrunuchime</span>を以下のように実行して下さい。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clrunuchime \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--contigmembers=*.contigmembers.gz \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--otufile=*.otu.gz \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
            </span>
</p>
<p class="ltx_p">旧パイプラインを使用した場合は<span class="ltx_text ltx_font_typewriter">--contigmembers</span>オプションを、新パイプラインを使用した場合は<span class="ltx_text ltx_font_typewriter">--otufile</span>オプションを使用します。
上記の例は一度に説明するためのもので、両方を指定してはいけません。
なお、どちらも指定しなかった場合、<span class="ltx_text ltx_font_typewriter">*.otu.gz</span>が入力ファイルと同じフォルダに存在すればそれを、<span class="ltx_text ltx_font_typewriter">*.contigmembers.gz</span>が存在すればそれを見つけ出して使用しますので、普段は何も指定しなくてもいいでしょう。</p>
</div>
</section>
<section id="Ch5.S4.SS2" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">5.4.2 </span>Reference-based chimera removal</h4>

<div id="Ch5.S4.SS2.p1" class="ltx_para">
<p class="ltx_p"><span class="ltx_text ltx_font_typewriter">clrunuchime</span>を以下のように実行して下さい。
リファレンスなしでのキメラ配列除去を行った場合は、<span class="ltx_text ltx_font_typewriter">nonchimeras.fasta</span>というファイルがあるはずですので、それを入力ファイルにすればいいでしょう。</p>
</div>
<div id="Ch5.S4.SS2.p2" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clrunuchime \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--referencedb=リファレンスデータベース名 \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
            </span>
</p>
</div>
<div id="Ch5.S4.SS2.p3" class="ltx_para">
<p class="ltx_p">UCHIME用のリファレンスデータベース名と内容は以下の通りです。</p>
</div>
<div id="Ch5.S4.SS2.p4" class="ltx_para">
<dl id="Ch5.I3" class="ltx_description">
<dt id="Ch5.I3.ix1"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text" style="font-size:90%;">rdpgoldv9</span>
              </span></dt>
<dd>
<div id="Ch5.I3.ix1.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">細菌16S用のRDP Goldデータベース(v9版)</span>
                </p>
</div>
</dd>
<dt id="Ch5.I3.ix2"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text" style="font-size:90%;">unite20160101</span>
              </span></dt>
<dd>
<div id="Ch5.I3.ix2.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">真菌ITS用のUNITEデータベース(20160101版)</span>
                </p>
</div>
</dd>
<dt id="Ch5.I3.ix3"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text" style="font-size:90%;">unite20160101untrim</span>
              </span></dt>
<dd>
<div id="Ch5.I3.ix3.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">真菌ITS用のUNITEデータベース(20160101トリミングなし版)</span>
                </p>
</div>
</dd>
<dt id="Ch5.I3.ix4"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text" style="font-size:90%;">unite20160101its1</span>
              </span></dt>
<dd>
<div id="Ch5.I3.ix4.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">真菌ITS用のUNITEデータベース(20160101ITS1版)</span>
                </p>
</div>
</dd>
<dt id="Ch5.I3.ix5"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text" style="font-size:90%;">unite20160101its2</span>
              </span></dt>
<dd>
<div id="Ch5.I3.ix5.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">真菌ITS用のUNITEデータベース(20160101ITS2版)</span>
                </p>
</div>
</dd>
</dl>
</div>
</section>
<section id="Ch5.S4.SS3" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">5.4.3 </span>About contents of output folder</h4>

<div id="Ch5.S4.SS3.p1" class="ltx_para">
<p class="ltx_p">出力フォルダに保存されるファイルは以下の4つです。</p>
</div>
<div id="Ch5.S4.SS3.p2" class="ltx_para">
<dl id="Ch5.I4" class="ltx_description">
<dt id="Ch5.I4.ix1"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text" style="font-size:90%;">chimeras.fasta</span>
              </span></dt>
<dd>
<div id="Ch5.I4.ix1.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">キメラと判定された配列</span>
                </p>
</div>
</dd>
<dt id="Ch5.I4.ix2"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text" style="font-size:90%;">nonchimeras.fasta</span>
              </span></dt>
<dd>
<div id="Ch5.I4.ix2.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">キメラでないと判定された配列</span>
                </p>
</div>
</dd>
<dt id="Ch5.I4.ix3"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text" style="font-size:90%;">uchimealns.txt</span>
              </span></dt>
<dd>
<div id="Ch5.I4.ix3.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">判定時のアライメント</span>
                </p>
</div>
</dd>
<dt id="Ch5.I4.ix4"><span class="ltx_tag ltx_tag_description">
                <span class="ltx_text" style="font-size:90%;">uchimeout.txt</span>
              </span></dt>
<dd>
<div id="Ch5.I4.ix4.p1" class="ltx_para">
<p class="ltx_p">
                  <span class="ltx_text" style="font-size:90%;">判定時の親配列やスコアなど</span>
                </p>
</div>
</dd>
</dl>
</div>
<div id="Ch5.S4.SS3.p3" class="ltx_para">
<p class="ltx_p"><span class="ltx_text ltx_font_typewriter">uchimeout.txt</span>の各項目の意味は
<br class="ltx_break"><a href="http://drive5.com/usearch/manual/uchimeout.html" title="" class="ltx_ref">http://drive5.com/usearch/manual/uchimeout.html</a>
<br class="ltx_break">をご覧下さい。</p>
</div>
</section>
</section>
<section id="Ch5.S5" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">5.5 </span>Excluding low-abundance OTUs from OTU sequences</h3>

<div id="Ch5.S5.p1" class="ltx_para">
<p class="ltx_p">クラスタリングの出力フォルダには、<span class="ltx_text ltx_font_typewriter">clustered.fasta.gz</span> (新パイプライン)または<span class="ltx_text ltx_font_typewriter">assembled.fasta</span> (旧パイプライン)というファイルとして、最終的に得られた代表配列もしくはコンセンサス配列が保存されています。
これを次章のDNAバーコーディングに用いますが、全サンプルを通して少量しか出現しないものまで含めると数が多すぎることがあります。
そこで、以下のようにして5配列以上出現するものだけを抽出することができます。
配列数の閾値を適当に変えてもいいでしょう。
この閾値は慎重に決定して下さい。
ノイジー配列・キメラ配列の除去を行なっている場合には必要ないかもしれません。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clfilterseq \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--contigmembers=*.contigmembers.gz \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--otufile=*.otu.gz \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minnseq=5 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile↓</span>
          </span>
</p>
<p class="ltx_p">旧パイプラインを使用した場合は<span class="ltx_text ltx_font_typewriter">--contigmembers</span>オプションを、新パイプラインを使用した場合は<span class="ltx_text ltx_font_typewriter">--otufile</span>オプションを使用します。
上記の例は一度に説明するためのもので、両方を指定してはいけません。
なお、どちらも指定しなかった場合、<span class="ltx_text ltx_font_typewriter">*.otu.gz</span>がinputfileと同じフォルダに存在すればそれを、<span class="ltx_text ltx_font_typewriter">*.contigmembers.gz</span>が存在すればそれを見つけ出して使用しますので、普段は何も指定しなくてもいいでしょう。
ここで作成したFASTA配列ファイルを第<a href="#Ch5.S2" title="5.2 Excluding specified OTUs and/or samples from summary table ‣ Chapter 5 Summarizing and post-processing of OTU picking results ‣ Matabarcoding and DNA barcoding for Ecologists" class="ltx_ref"><span class="ltx_text ltx_ref_tag">5.2</span></a>節の<span class="ltx_text ltx_font_typewriter">clfiltersum</span>の<span class="ltx_text ltx_font_typewriter">--otuseq</span>オプションに指定して集計表を加工することで、この配列ファイルから除去されたOTUを集計表からも除去することができます。</p>
</div>
</section>
<section id="Ch5.S6" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">5.6 </span>Sequence splitting based on conservative motif recognition</h3>

<div id="Ch5.S6.p1" class="ltx_para">
<p class="ltx_p">ここまでの配列が複数領域にまたがっている場合(たとえばITS1・5.8S rRNA・ITS2といった風に)、領域ごとに分割した方が良いかもしれません。
領域間かその付近に保存的な領域があれば、それを境界の目印として領域を分割できます。
特にユニバーサルプライマー配列などを使うといいでしょう。
これは以下のコマンドで行うことができます。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; cldivseq \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--query=保存領域の配列 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--border=start \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile1 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile2↓</span>
          </span>
</p>
<p class="ltx_p">このコマンドでは、指定された配列をNeedleman-Wunschアルゴリズムを用いたアライメントによって15%まで不一致を許して探します。
該当する部位が見つかったら、その左端を境界として配列を2つに分け、それぞれ出力ファイル1と出力ファイル2に保存します。
該当する部位が見つからなかった場合は、そのまま出力ファイル1だけに保存されます。
信頼度配列ファイルがある場合には、こちらも塩基配列に合わせて分割されます。
なお、指定する配列は対象配列と同一のストランドにしておいて下さい。
ストランドが異なる場合は、<span class="ltx_text ltx_font_typewriter">--reversecomplement</span>オプションを付加して実行することで、<span class="ltx_text ltx_font_typewriter">--query</span>に指定された配列の逆相補配列をアライメントに用いるようになります。</p>
</div>
<div id="Ch5.S6.p2" class="ltx_para">
<p class="ltx_p"><span class="ltx_text ltx_font_typewriter">--border</span>オプションを<span class="ltx_text ltx_font_typewriter">start</span>ではなく<span class="ltx_text ltx_font_typewriter">end</span>にすることで、検索に一致した部位の右端を境界として分割することができます。
<span class="ltx_text ltx_font_typewriter">--border=both</span>にした場合は、検索に一致した部位はどちらのファイルにも出力されません(これがデフォルト設定です)。
検索した配列が見つからなかった場合は出力ファイル1のみに配列が保存され、出力ファイル2には配列が保存されませんが、これでは都合が悪い場合は<span class="ltx_text ltx_font_typewriter">--makedummy</span>オプションを付加することでダミーの塩基配列が出力ファイル2に保存されます。
分割した各領域の配列で別個にホスト生物を同定した上で同定結果を統合する場合など、配列間で対応がとれていないと困る場合にご利用下さい。
ここでは2分割の例を挙げましたが、分割後にさらに分割を繰り返すことで3分割や4分割も可能です。</p>
</div>
</section>
<section id="Ch5.S7" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">5.7 </span>ITS, SSU rRNA or LSU rRNA sequence extraction using ITSx or Metaxa</h3>

<div id="Ch5.S7.p1" class="ltx_para">
<p class="ltx_p">ITSxは核ITS1・ITS2領域を認識してその部位だけを取り出すことができるプログラムです<cite class="ltx_cite"><a href="#bib.bib4" title="" class="ltx_ref">(Bengtsson-Palme <span class="ltx_text ltx_font_italic">et al.</span>, 2013)</a></cite>。
多くの分類群でITSは非常に変異に富んでいますが、ずっと保守的なSSU・LSU rRNAと隣接しているため、これらの配列が残っていると、その配列の影響でITSでは明確に区別できる遠縁な生物のSSU・LSU配列がBLAST検索時にヒットしてしまい、うまく同定できなくなってしまうことがあります。
ITSxでITS領域のみを切り出して同定することでこのような問題に対処できます。</p>
</div>
<div id="Ch5.S7.p2" class="ltx_para">
<p class="ltx_p">MetaxaはITSxと同様にSSU (12S/16S/18S) rRNAを区別してそれぞれの部位だけを取り出すことができるプログラムです<cite class="ltx_cite"><a href="#bib.bib3" title="" class="ltx_ref">(Bengtsson <span class="ltx_text ltx_font_italic">et al.</span>, 2011)</a></cite>。
SSU rRNAは真核生物のメタバーコーディング・DNAバーコーディングに広く利用されている領域ですが、核だけでなくミトコンドリアや葉緑体、混入した細菌にも含まれているため、核のSSUを標的とするPCRを行った場合にも多少はミトコンドリアや細菌のSSU rRNAが混入してしまいます。
そのような配列は群集生態学的解析の障害となるため、このプログラムで前もって除去しておくと後の解析が楽になります。</p>
</div>
<div id="Ch5.S7.p3" class="ltx_para">
<p class="ltx_p">いずれのプログラムもインストール方法、使用方法をここでは説明しません。
それぞれのWebサイトとマニュアルをご覧下さい。
これらのプログラムで作成したFASTA配列ファイルを第<a href="#Ch5.S2" title="5.2 Excluding specified OTUs and/or samples from summary table ‣ Chapter 5 Summarizing and post-processing of OTU picking results ‣ Matabarcoding and DNA barcoding for Ecologists" class="ltx_ref"><span class="ltx_text ltx_ref_tag">5.2</span></a>節の<span class="ltx_text ltx_font_typewriter">clfiltersum</span>の<span class="ltx_text ltx_font_typewriter">--otuseq</span>オプションに指定して集計表を加工することで、この配列ファイルに含まれているOTUだけの集計表を作成することができます。</p>
</div>
</section>
<section id="Ch5.S8" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">5.8 </span>Searching and excluding non-target sequences</h3>

<div id="Ch5.S8.p1" class="ltx_para">
<p class="ltx_p">ITSxやMetaxaはITSとSSU rRNA領域にしか使用できませんが、もっと汎用的な非ターゲット領域の探索方法があります。
一つはBLASTなどで検索して配列がどの遺伝子のものかを判定してくれるプログラムにかけることで、もう一つは多重整列プログラムを利用することです。
ClustalW2・ClustalX2 <cite class="ltx_cite"><a href="#bib.bib13" title="" class="ltx_ref">(Larkin <span class="ltx_text ltx_font_italic">et al.</span>, 2007)</a></cite>やMAFFT <cite class="ltx_cite"><a href="#bib.bib10" title="" class="ltx_ref">(Katoh &amp; Standley, 2013)</a></cite>には、多重整列した結果を系統的に近いものが近くになるように並び替えて出力することができます。
この機能を使って並び替えた上で、多重整列の閲覧ソフトで表示して目で確認すればどれが非ターゲット領域かすぐわかります。
非ターゲット領域のOTUを除去したFASTA配列ファイルを第<a href="#Ch5.S2" title="5.2 Excluding specified OTUs and/or samples from summary table ‣ Chapter 5 Summarizing and post-processing of OTU picking results ‣ Matabarcoding and DNA barcoding for Ecologists" class="ltx_ref"><span class="ltx_text ltx_ref_tag">5.2</span></a>節の<span class="ltx_text ltx_font_typewriter">clfiltersum</span>の<span class="ltx_text ltx_font_typewriter">--otuseq</span>オプションに指定して集計表を加工することで、この配列ファイルに含まれているOTUだけの集計表を作成することができます。</p>
</div>
</section>
</section>
<section id="Ch6" class="ltx_chapter">
<h2 class="ltx_title ltx_title_chapter">
<span class="ltx_tag ltx_tag_chapter">Chapter 6 </span>Data deposition to DRA</h2>

<div id="Ch6.p1" class="ltx_para">
<p class="ltx_p">クラスタリング後の配列は従来通りDDBJなどに登録すればいいですが、新型シーケンサーの生データはDDBJ Sequence Read Archive (DRA)というところへ登録することになっています(サンガー法シーケンサーの生データはTrace Archive)。
本書にあるように複数サンプルからのDNAにタグを付けてシーケンスした場合、サンプルごとのデータファイルに分割して登録することを求められます。
これは<span class="ltx_text ltx_font_typewriter">clsplitseq</span>によって分割したFASTQを使えば問題ありません。
同一鋳型由来配列ファイルが複数ある場合は、GZIP圧縮を解除してから連結して再圧縮するだけで構いません。</p>
</div>
<div id="Ch6.p2" class="ltx_para">
<p class="ltx_p">また、サンプルについての情報(メタデータ)を記述したXMLファイルを作成しなくてはなりません。
データ登録を受け付けている側でも、XML作成を補助するツールが用意されていますが大量サンプルを扱っている場合は用意するのは大変です。
そこで、DRA登録用XML作成を補助する<span class="ltx_text ltx_font_typewriter">clmaketsv</span>および<span class="ltx_text ltx_font_typewriter">clmakexml</span>というコマンドを用意しています。
以下でこれらの使用法を説明します。</p>
</div>
<div id="Ch6.p3" class="ltx_para">
<p class="ltx_p">DRAへの登録には、DDBJが管理するD-wayでのユーザーアカウント作成と公開鍵の登録が必要です。
詳しくはDRA Handbook
<br class="ltx_break"><a href="http://trace.ddbj.nig.ac.jp/dra/submission.shtml" title="" class="ltx_ref">http://trace.ddbj.nig.ac.jp/dra/submission.shtml</a>
<br class="ltx_break">をご覧下さい。
登録の際、Submission、Study、Experiment、Sample、Runの概念と対応関係を把握している必要がありますので、その部分を特に注意して読んで理解しておいて下さい。</p>
</div>
<div id="Ch6.p4" class="ltx_para">
<p class="ltx_p">DRAでは、StudyはBioProjectという研究プロジェクトデータベースに登録して、それを参照することになります。
BioProjectへの登録はBioProject Handbook
<br class="ltx_break"><a href="http://trace.ddbj.nig.ac.jp/bioproject/submission.html" title="" class="ltx_ref">http://trace.ddbj.nig.ac.jp/bioproject/submission.html</a>
<br class="ltx_break">を参照して予め済ませておいて下さい。</p>
</div>
<div id="Ch6.p5" class="ltx_para">
<p class="ltx_p">さらに、SampleはBioSampleという研究サンプルデータベースに登録して、それを参照します。
BioSampleへの登録はBioSample Handbook
<br class="ltx_break"><a href="http://trace.ddbj.nig.ac.jp/biosample/submission.html" title="" class="ltx_ref">http://trace.ddbj.nig.ac.jp/biosample/submission.html</a>
<br class="ltx_break">を参照して予め済ませておいて下さい。
なお、メタゲノムを鋳型としてユニバーサルプライマーで増幅を行ったシーケンスサンプルの場合、MIMarks-SurveyをMIxSのタイプとして指定します。
サンプルの採集された高度、水深、温度、湿度、pHなど、様々な情報を付加しておくことができます。
後世のためにも、面倒がらずにできるだけ多くの情報を付加しておいて下さい。
なお、各項目の意味は、Genomic Standards Consortiumから提供されているチェックリスト
<br class="ltx_break"><a href="http://wiki.gensc.org/index.php?title=MIMARKS" title="" class="ltx_ref">http://wiki.gensc.org/index.php?title=MIMARKS</a>
<br class="ltx_break">を確認して下さい。
これを見てもわからない場合はDDBJに問い合わせて下さい。
特に、地下や海底下のサンプルでは、平均海水面からサンプリング地点までの高さ・深さ、地面からサンプリング地点までの高さ・深さ、平均海水面から地面までの高さ・深さを入れる項目があってややこしいのでご注意下さい。</p>
</div>
<section id="Ch6.S1" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">6.1 </span>Preparing tab-delimited text file for XML generation</h3>

<div id="Ch6.S1.p1" class="ltx_para">
<p class="ltx_p">DRAに登録するXMLには、登録するFASTQファイルごとにそのファイルに含まれている配列の情報を記述します。
これは非常に複雑で大変なため、一旦単純なタブ区切りテキストに書き出した上で、Excelなどを用いてそれを編集し、編集後のタブ区切りテキストからXMLを作成します。
タブ区切りテキストを生成するコマンドは<span class="ltx_text ltx_font_typewriter">clmaketsv</span>です。
以下のように使って下さい。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clmaketsv \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile1 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">中略 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfileN \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile↓</span>
          </span>
</p>
<p class="ltx_p">入力ファイルは登録するものを指定して下さい。
ワイルドカードも使えます。
タブ区切りテキストができたら、表計算ソフトなどに読み込ませて、各セルを埋めていって下さい。
なお、Excelでは特定の文字列を勝手に日時などと認識して変換してしまう機能(無効にする方法はありません)がありますので、十分注意して下さい。
各セル内では、<span class="ltx_text ltx_font_typewriter">[Foo,Bar]</span>とあった場合は<span class="ltx_text ltx_font_typewriter">Foo</span>または<span class="ltx_text ltx_font_typewriter">Bar</span>のどちらかを選んで括弧と選ばなかったものを消して下さい。
<span class="ltx_text ltx_font_typewriter">&lt;Fill in this cell&gt;</span>とあった場合は、<span class="ltx_text ltx_font_typewriter">&lt;&gt;</span>内の指示に従ってセルを埋めて下さい(括弧は残さない)。
その他のセルは適当に自分で考えて何とかして下さい。
なお、BioProject IDやBioSample IDは、正式なアクセッション番号がまだ発行されていない場合、DDBJに申請したときのSubmission ID (BioProjectではPSUBから始まり、BioSampleではSSUBから始まるもの)を入力しておけば問題ありません。</p>
</div>
</section>
<section id="Ch6.S2" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">6.2 </span>XML generation from tab-delimited text</h3>

<div id="Ch6.S2.p1" class="ltx_para">
<p class="ltx_p">タブ区切りテキストの編集が終わったら、タブ区切りテキストとして保存した上で以下のように<span class="ltx_text ltx_font_typewriter">clmakexml</span>を実行して下さい。
登録に必要なXMLファイル群が生成されます。
なお、タブ区切りテキストは一度に複数指定することも可能です。
ただし、各ファイルの最初の3行に記述する内容は、最初のファイルのものしか利用されません。
2ファイル目以降の最初の3行は無視されます。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clmakexml \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">タブ区切りテキストファイル \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">DRAから割り振られたsubmission-ID↓</span>
          </span>
</p>
<p class="ltx_p">DRAでは、「Create new submission(s)」によってsubmission-IDが割り振られます。
<span class="ltx_text ltx_font_typewriter">ユーザーID-0001</span>などとなっているはずです。
<span class="ltx_text ltx_font_typewriter">DRAから割り振られたsubmission-ID</span>にはこれを指定します。
すると、<span class="ltx_text ltx_font_typewriter">submission-ID.*.xml</span>という名前のファイルが3つ生成されます。
これをDRAのXML Upload機能を利用して送信します。
その他もろもろの処理が終わると、登録してあるメールアドレスに割り振られたアクセッション番号が送られてきます。
ただし、論文にはBioProject IDを書くことが多いと思います。</p>
</div>
</section>
</section>
<section id="Ch7" class="ltx_chapter">
<h2 class="ltx_title ltx_title_chapter">
<span class="ltx_tag ltx_tag_chapter">Chapter 7 </span>Estimation of host organisms of nucleotide sequences (DNA barcoding)</h2>

<div id="Ch7.p1" class="ltx_para">
<p class="ltx_p">DNAバーコーディングは、近年非常に多くの分野で応用が進められている、生物の同定方法です。
しかし、既知配列データベースが不十分な上、それを前提とした同定アルゴリズムが欠けていました。
そこで、筆者は「問い合わせ配列と最近隣配列との間の変異量＜分類群内の最大変異量」という規準を考案し、これを実現するアルゴリズムQCauto法<cite class="ltx_cite"><a href="#bib.bib18" title="" class="ltx_ref">(Tanabe &amp; Toju, 2013)</a></cite>をClaidentに実装しました。
ただし、その配列のホストの可能性がある全生物が記載済みで、しかもそのバーコード配列もデータベースに登録済みである場合には、そのような難しいことを考えずに最近隣配列と同種とみなせばよいでしょう。
そちらの方法もClaidentには実装してあります。
以下のコマンドは全てターミナルかコンソールで実行して下さい。
基本的なターミナルの使い方の知識は持っているものとして話を進めます。</p>
</div>
<section id="Ch7.S1" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">7.1 </span>Retrieval of neighborhood sequences based on BLAST search</h3>

<div id="Ch7.S1.p1" class="ltx_para">
<p class="ltx_p">以下のコマンドで、「問い合わせ配列と最近隣配列との間の変異量＜分類群内の最大変異量」という条件を満たすために検討すべき近隣既知配列群(のGenBank ID)を取得できます。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clidentseq \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--blastdb=overall_genus \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile↓</span>
          </span>
</p>
<p class="ltx_p">入力ファイルにはFASTA形式の塩基配列ファイルを指定します。
<span class="ltx_text ltx_font_typewriter">--blastdb</span>オプションでは、BLAST検索に用いるデータベース名を指定します。
筆者が用意しているデータベースは以下の通りです。</p>
<dl id="Ch7.I1" class="ltx_description">
<dt id="Ch7.I1.ix1"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">animals_COX1_genus</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix1.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">動物(Metazoa)のmtDNA </span>
                <span class="ltx_text ltx_font_italic" style="font-size:90%;">COX1</span>
                <span class="ltx_text" style="font-size:90%;">配列で属以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix2"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">animals_COX1_species</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix2.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">同上だが種以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix3"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">animals_mt_genus</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix3.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">動物(Metazoa)のmtDNA配列で属以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix4"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">animals_mt_species</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix4.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">同上だが種以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix5"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">eukaryota_LSU_genus</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix5.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">真核生物のLSU (28S) rRNA配列で属以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix6"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">eukaryota_LSU_species</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix6.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">同上だが種以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix7"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">eukaryota_SSU_genus</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix7.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">真核生物のSSU (18S) rRNA配列で属以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix8"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">eukaryota_SSU_species</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix8.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">同上だが種以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix9"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">fungi_ITS_genus</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix9.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">真菌のITS配列で属以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix10"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">fungi_ITS_species</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix10.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">同上だが種以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix11"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">overall_class</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix11.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">NCBI ntの中で綱以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix12"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">overall_order</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix12.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">同上だが目以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix13"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">overall_family</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix13.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">同上だが科以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix14"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">overall_genus</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix14.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">同上だが属以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix15"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">overall_species</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix15.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">同上だが種以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix16"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">plants_matK_genus</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix16.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">緑色植物のcpDNA </span>
                <span class="ltx_text ltx_font_italic" style="font-size:90%;">matK</span>
                <span class="ltx_text" style="font-size:90%;">配列で属以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix17"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">plants_matK_species</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix17.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">同上だが種以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix18"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">plants_rbcL_genus</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix18.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">緑色植物のcpDNA </span>
                <span class="ltx_text ltx_font_italic" style="font-size:90%;">rbcL</span>
                <span class="ltx_text" style="font-size:90%;">配列で属以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix19"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">plants_rbcL_species</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix19.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">同上だが種以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix20"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">plants_trnH-psbA_genus</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix20.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">緑色植物のcpDNA </span>
                <span class="ltx_text ltx_font_italic" style="font-size:90%;">trnH</span>
                <span class="ltx_text" style="font-size:90%;">–</span>
                <span class="ltx_text ltx_font_italic" style="font-size:90%;">psbA</span>
                <span class="ltx_text" style="font-size:90%;">配列で属以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix21"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">plants_trnH-psbA_species</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix21.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">同上だが種以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix22"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">prokaryota_16S_genus</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix22.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">原核生物の16S rRNA配列で属以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix23"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">prokaryota_16S_species</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix23.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">同上だが種以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix24"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">prokaryota_all_genus</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix24.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">原核生物の配列で属以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix25"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">prokaryota_all_species</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix25.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text" style="font-size:90%;">同上だが種以下の情報があるもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix26"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">semiall_class</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix26.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">overall_class</span>
                <span class="ltx_text" style="font-size:90%;">から脊椎動物と</span>
                <span class="ltx_text ltx_font_italic" style="font-size:90%;">Caenorhabditis</span>
                <span class="ltx_text" style="font-size:90%;">と</span>
                <span class="ltx_text ltx_font_italic" style="font-size:90%;">Drosophila</span>
                <span class="ltx_text" style="font-size:90%;">を除いたもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix27"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">semiall_order</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix27.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">overall_order</span>
                <span class="ltx_text" style="font-size:90%;">から脊椎動物と</span>
                <span class="ltx_text ltx_font_italic" style="font-size:90%;">Caenorhabditis</span>
                <span class="ltx_text" style="font-size:90%;">と</span>
                <span class="ltx_text ltx_font_italic" style="font-size:90%;">Drosophila</span>
                <span class="ltx_text" style="font-size:90%;">を除いたもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix28"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">semiall_family</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix28.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">overall_family</span>
                <span class="ltx_text" style="font-size:90%;">から脊椎動物と</span>
                <span class="ltx_text ltx_font_italic" style="font-size:90%;">Caenorhabditis</span>
                <span class="ltx_text" style="font-size:90%;">と</span>
                <span class="ltx_text ltx_font_italic" style="font-size:90%;">Drosophila</span>
                <span class="ltx_text" style="font-size:90%;">を除いたもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix29"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">semiall_genus</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix29.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">overall_genus</span>
                <span class="ltx_text" style="font-size:90%;">から脊椎動物と</span>
                <span class="ltx_text ltx_font_italic" style="font-size:90%;">Caenorhabditis</span>
                <span class="ltx_text" style="font-size:90%;">と</span>
                <span class="ltx_text ltx_font_italic" style="font-size:90%;">Drosophila</span>
                <span class="ltx_text" style="font-size:90%;">を除いたもの</span>
              </p>
</div>
</dd>
<dt id="Ch7.I1.ix30"><span class="ltx_tag ltx_tag_description">
              <span class="ltx_text" style="font-size:90%;">semiall_species</span>
            </span></dt>
<dd>
<div id="Ch7.I1.ix30.p1" class="ltx_para">
<p class="ltx_p">
                <span class="ltx_text ltx_font_typewriter" style="font-size:90%;">overall_species</span>
                <span class="ltx_text" style="font-size:90%;">から脊椎動物と</span>
                <span class="ltx_text ltx_font_italic" style="font-size:90%;">Caenorhabditis</span>
                <span class="ltx_text" style="font-size:90%;">と</span>
                <span class="ltx_text ltx_font_italic" style="font-size:90%;">Drosophila</span>
                <span class="ltx_text" style="font-size:90%;">を除いたもの</span>
              </p>
</div>
</dd>
</dl>
<p class="ltx_p"><span class="ltx_text ltx_font_typewriter">overall_*</span>は非常に巨大なため、多くのメモリを必要としますが、これらは万能なのでどんな配列にも使えますし、想定外の分類群や想定外の遺伝子座の配列の混入があってもとんでもない誤同定を避けることができます。
<span class="ltx_text ltx_font_typewriter">overall_genus</span>では属以下の情報がある、つまりそれなりに分類群の情報が信頼できそうな配列を選別していますが、あまりにマイナーな分類群だとそれでは近隣配列が見つからないことがあります。
その場合は、綱以下まで同定された配列のデータベース<span class="ltx_text ltx_font_typewriter">overall_class</span>を使ってみて下さい。
その他のデータベースは、<span class="ltx_text ltx_font_typewriter">overall_*</span>よりもメモリ占有量がずっと少ないため、メモリの少ないコンピュータでも動作します。</p>
</div>
<section id="Ch7.S1.SS1" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">7.1.1 </span>Accelerating BLAST search using cache databases</h4>

<div id="Ch7.S1.SS1.p1" class="ltx_para">
<p class="ltx_p"><span class="ltx_text ltx_font_typewriter">clidentseq</span>はBLAST検索を何度も繰り返しますが、それなりの時間と大量のメモリを必要とします。そこで、予め問い合わせ配列と類似した配列の上位1万本(設定で変更可)のヒットした部位のみをBLASTデータベースから取得してキャッシュとなるBLASTデータベースを作成しておき、<span class="ltx_text ltx_font_typewriter">clidentseq</span>ではこれを対象として検索することで、劇的な高速化が可能です。これを行うのが<span class="ltx_text ltx_font_typewriter">clmakecachedb</span>で、以下のように用います。</p>
</div>
<div id="Ch7.S1.SS1.p2" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clmakecachedb \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--blastdb=overall_genus \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfolder↓</span>
            </span>
</p>
</div>
<div id="Ch7.S1.SS1.p3" class="ltx_para">
<p class="ltx_p"><span class="ltx_text ltx_font_typewriter">clidentseq</span>の実行時には、<span class="ltx_text ltx_font_typewriter">--blastdb</span>オプションに上記コマンドの出力フォルダを指定することで、キャッシュデータベースが使われます。
ただし、<span class="ltx_text ltx_font_typewriter">clmakecachedb</span>と<span class="ltx_text ltx_font_typewriter">clidentseq</span>の入力ファイルは同一である必要があります。
メモリの使用量も大幅に減少しますので、<span class="ltx_text ltx_font_typewriter">overall_*</span>を使用してもメモリ16GB程度で済みます。</p>
</div>
</section>
<section id="Ch7.S1.SS2" class="ltx_subsection">
<h4 class="ltx_title ltx_title_subsection">
<span class="ltx_tag ltx_tag_subsection">7.1.2 </span>In the case of perfect reference database</h4>

<div id="Ch7.S1.SS2.p1" class="ltx_para">
<p class="ltx_p">問い合わせ配列のホストの可能性がある全生物が記載済みで、しかもそのバーコード配列も全て参照配列データベースに登録済みである場合には、以下のようにして類似度99%以上の上位1位(1位タイを含む)を近隣配列とすればいいでしょう。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clidentseq \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">blastn -task megablast -word_size 16 end \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--method=1,99% \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--blastdb=overall_genus \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--numthreads=NumberOfCPUs \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
            </span>
<span class="ltx_p ltx_align_left">
              <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile↓</span>
            </span>
</p>
</div>
<div id="Ch7.S1.SS2.p2" class="ltx_para">
<p class="ltx_p">類似度99%以上の上位1位の配列を探し出す場合、BLAST検索オプションは<span class="ltx_text ltx_font_typewriter">-task megablast -word_size 16</span>としても見逃すことはまずないでしょうから、ここでは高速化のためにこのように指定しています。</p>
</div>
</section>
</section>
<section id="Ch7.S2" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">7.2 </span>Taxonomic assignment based on neighborhood sequences</h3>

<div id="Ch7.S2.p1" class="ltx_para">
<p class="ltx_p">以下のコマンドで、近隣既知配列群の所属分類群が同一になるまで分類階層を上げていくことで配列を同定します。
これはlowest common ancestor (LCA) algorithmと呼ばれています<cite class="ltx_cite"><a href="#bib.bib8" title="" class="ltx_ref">(Huson <span class="ltx_text ltx_font_italic">et al.</span>, 2007)</a></cite>。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; classigntax \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--taxdb=overall_genus \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile↓</span>
          </span>
</p>
<p class="ltx_p">inputfileには前節で作成した<span class="ltx_text ltx_font_typewriter">clidentseq</span>の出力ファイルを指定して下さい。
<span class="ltx_text ltx_font_typewriter">--taxdb</span>は既知配列の所属分類群データベース指定オプションです。
BLASTデータベースと同じ名前のものがインストールされていますので、それを指定して下さい。</p>
</div>
<div id="Ch7.S2.p2" class="ltx_para">
<p class="ltx_p"><span class="ltx_text ltx_font_typewriter">classigntax</span>のデフォルト設定では、少なくとも2本以上の近隣既知配列がないと同定することができません。
第<a href="#Ch7.S1.SS2" title="7.1.2 In the case of perfect reference database ‣ 7.1 Retrieval of neighborhood sequences based on BLAST search ‣ Chapter 7 Estimation of host organisms of nucleotide sequences (DNA barcoding) ‣ Matabarcoding and DNA barcoding for Ecologists" class="ltx_ref"><span class="ltx_text ltx_ref_tag">7.1.2</span></a>節のように<span class="ltx_text ltx_font_typewriter">clidentseq</span>の実行時に<span class="ltx_text ltx_font_typewriter">--method=1,99%</span>を指定していると、近隣既知配列が1本しかないために一つも同定できないことになります。
この場合は以下のようにして必要な近隣既知配列数を1にします。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; classigntax \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--taxdb=overall_genus \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minnsupporter=1 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile↓</span>
          </span>
</p>
<p class="ltx_p">出力ファイルは表<a href="#Ch7.T1" title="Table  7.1 ‣ 7.2 Taxonomic assignment based on neighborhood sequences ‣ Chapter 7 Estimation of host organisms of nucleotide sequences (DNA barcoding) ‣ Matabarcoding and DNA barcoding for Ecologists" class="ltx_ref"><span class="ltx_text ltx_ref_tag">7.1</span></a>のような形のタブ区切りのテキストになっています。</p>
</div>
<figure id="Ch7.T1" class="ltx_table">
<table class="ltx_tabular ltx_centering ltx_align_middle">
<thead class="ltx_thead">
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">query</span>
              </th>
<th class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">phylum</span>
              </th>
<th class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">genus</span>
              </th>
<th class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">species</span>
              </th>
</tr>
</thead>
<tbody class="ltx_tbody">
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r ltx_border_tt">
                <span class="ltx_text" style="font-size:80%;">seqA</span>
              </th>
<td class="ltx_td ltx_align_left ltx_border_tt">
                <span class="ltx_text" style="font-size:80%;">Ascomycota</span>
              </td>
<td class="ltx_td ltx_align_left ltx_border_tt">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Chloridium</span>
              </td>
<td class="ltx_td ltx_align_left ltx_border_tt">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Chloridium virescens</span>
              </td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">seqB</span>
              </th>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">Ascomycota</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Chloridium</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Chloridium virescens</span>
              </td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">seqC</span>
              </th>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">Ascomycota</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Chloridium</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Chloridium virescens</span>
              </td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">seqD</span>
              </th>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">Basidiomycota</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Amanita</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Amanita fuliginea</span>
              </td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">seqE</span>
              </th>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">Basidiomycota</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Coltriciella</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Coltriciella dependens</span>
              </td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">seqF</span>
              </th>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">Basidiomycota</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Filobasidium</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Filobasidium uniguttulatum</span>
              </td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">seqG</span>
              </th>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">Basidiomycota</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Laccaria</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Laccaria bicolor</span>
              </td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">seqH</span>
              </th>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">Basidiomycota</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Lactarius</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Lactarius quietus</span>
              </td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">seqI</span>
              </th>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">Basidiomycota</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Russula</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Russula densifolia</span>
              </td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">seqJ</span>
              </th>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">Basidiomycota</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Russula</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Russula densifolia</span>
              </td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">seqK</span>
              </th>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">Basidiomycota</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Russula</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Russula densifolia</span>
              </td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">seqL</span>
              </th>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">Basidiomycota</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Russula</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Russula vesca</span>
              </td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">seqM</span>
              </th>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">Basidiomycota</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Agaricus</span>
              </td>
<td class="ltx_td"></td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">seqN</span>
              </th>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">Basidiomycota</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Amanita</span>
              </td>
<td class="ltx_td"></td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">seqO</span>
              </th>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">Basidiomycota</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Amanita</span>
              </td>
<td class="ltx_td"></td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">seqP</span>
              </th>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">Ascomycota</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Bisporella</span>
              </td>
<td class="ltx_td"></td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">seqQ</span>
              </th>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">Ascomycota</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Capronia</span>
              </td>
<td class="ltx_td"></td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">seqR</span>
              </th>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">Ascomycota</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Capronia</span>
              </td>
<td class="ltx_td"></td>
</tr>
<tr class="ltx_tr">
<th class="ltx_td ltx_align_left ltx_border_r">
                <span class="ltx_text" style="font-size:80%;">seqS</span>
              </th>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text" style="font-size:80%;">Ascomycota</span>
              </td>
<td class="ltx_td ltx_align_left">
                <span class="ltx_text ltx_font_italic" style="font-size:80%;">Cenococcum</span>
              </td>
<td class="ltx_td"></td>
</tr>
</tbody>
</table>
<figcaption class="ltx_caption"><span class="ltx_tag ltx_tag_table">Table  7.1: </span>同定結果の例 — 空欄はunidentified、つまり不明ということです。
横幅を抑えるためいくつかの分類階層は省いてあります。</figcaption>
</figure>
<div id="Ch7.S2.p3" class="ltx_para">
<p class="ltx_p"><span class="ltx_text ltx_font_typewriter">classigntax</span>のデフォルト設定では、全ての近隣既知配列の所属分類群が同一になるまで分類階層を上げていきますので、誤同定された配列が混ざっていたりした場合には下位の分類階層はほとんど不明になってしまいます。
これは「ほぼ正しいと思われる」結果を得る＝「間違えない」ことをデフォルト設定では優先しているためですが、近隣既知配列が多い場合、多少不一致を許容して誤同定の可能性を増やしてでもできるだけ下位の分類階層まで同定したい場合があります。
そのような場合には以下のように<span class="ltx_text ltx_font_typewriter">--maxpopposer</span>と<span class="ltx_text ltx_font_typewriter">--minsoratio</span>を指定します。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; classigntax \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--taxdb=overall_genus \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--maxpopposer=0.05 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--minsoratio=19 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">inputfile \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile↓</span>
          </span>
</p>
<p class="ltx_p"><span class="ltx_text ltx_font_typewriter">classigntax</span>は結果として採用する分類群に該当する配列をsupporter配列とし、該当しない配列をopposer配列とします。
<span class="ltx_text ltx_font_typewriter">--maxpopposer</span>にはopposer配列の存在を許容する割合を百分率で指定します。
<span class="ltx_text ltx_font_typewriter">--minsoratio</span>はsupporter配列数とopposer配列数の比の下限値を指定します。
supporter配列数とopposer配列数がこれを下回るような同定結果は許容せず分類階層を上げていきます。
2つのオプションが必要なのは、その分類階層の情報がない配列が存在し得るためです。
そのような配列はsupporterでもopposerでもないので、2つのオプションが必要になります。
上記の例では、近隣既知配列中、5%までのopposer配列を許容し、opposer配列の19倍以上のsupporter配列を必要とします。</p>
</div>
</section>
<section id="Ch7.S3" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">7.3 </span>Merging multiple taxonomic assignments based on consensus</h3>

<div id="Ch7.S3.p1" class="ltx_para">
<p class="ltx_p">例えば植物の<span class="ltx_text ltx_font_italic">rbcL</span>と<span class="ltx_text ltx_font_italic">matK</span>など、複数のバーコード領域を併用しなければ同定できないこともありますし、<span class="ltx_text ltx_font_typewriter">overall_genus</span>と<span class="ltx_text ltx_font_typewriter">overall_class</span>での同定結果のいいとこ取りをしたい場合があります。
また、厳密なLCAによる同定結果と5%までopposerを許容するLCAの同定結果の組み合わせでいいとこ取りしたい場合もあるでしょう。
そのようなことが、複数の同定結果を統合することで可能になります。</p>
</div>
<div id="Ch7.S3.p2" class="ltx_para">
<p class="ltx_p">植物の<span class="ltx_text ltx_font_italic">rbcL</span>と<span class="ltx_text ltx_font_italic">matK</span>を併用して同定する場合は、より深くまで同定できている方の結果を採用するのがいいでしょう。
これは以下のコマンドでできます。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clmergeassign \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--priority=equal \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--preferlower \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">rbcLでの同定結果 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">matKでの同定結果 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile↓</span>
          </span>
</p>
<p class="ltx_p">より保守的に考えるなら、以下のようにして、両方の同定結果が同一か、一方では未同定の場合だけ採用することもできます。
ただし、上の階層で不一致だった場合には、同定結果が一方で未同定でも採用されません。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clmergeassign \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--priority=equal \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">rbcLでの同定結果 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">matKでの同定結果 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile↓</span>
          </span>
</p>
<p class="ltx_p"><span class="ltx_text ltx_font_italic">trnH</span>–<span class="ltx_text ltx_font_italic">psbA</span>も併用して、最も深くまで同定できているものを採用する場合は以下のようにします。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clmergeassign \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--priority=equal \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--preferlower \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">rbcLでの同定結果 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">matKでの同定結果 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">trnH-psbAでの同定結果 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile↓</span>
          </span>
</p>
<p class="ltx_p"><span class="ltx_text ltx_font_typewriter">overall_genus</span>の結果を優先的に採用し、<span class="ltx_text ltx_font_typewriter">overall_genus</span>で同定できている分類階層までは一致しているが<span class="ltx_text ltx_font_typewriter">overall_class</span>の方がより下位まで同定できている場合だけ採用するには、以下のようにします。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clmergeassign \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--priority=descend \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">overall_genusでの同定結果 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">overall_classでの同定結果 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile↓</span>
          </span>
</p>
<p class="ltx_p">同様に、厳密なLCAによる結果を優先し、厳密なLCAで同定できている分類階層まで一致しているが制約を緩めたLCAによる結果の方がより下位まで同定できている場合にのみ採用するには、以下のようにします。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; clmergeassign \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">--priority=descend \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">厳密なLCAでの同定結果 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">制約を緩めたLCAでの同定結果 \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">outputfile↓</span>
          </span>
</p>
</div>
</section>
</section>
<section id="bib" class="ltx_bibliography">
<h2 class="ltx_title ltx_title_bibliography">References</h2>

<ul class="ltx_biblist" style="padding:0em 0em 0em 1em;text-indent:-1em;">
      
<li id="bib.bib2" class="ltx_bibitem" style="padding:0em 0em 0.2em 0em;">
        
        
        
        
        
<span class="ltx_bibblock">
Andrews, S. (2010) Software distributed by the author at
http://www.bioinformatics.babraham.ac.uk/projects/fastqc/.

</span>
      </li>
      
<li id="bib.bib3" class="ltx_bibitem" style="padding:0em 0em 0.2em 0em;">
        
        
        
        
        
<span class="ltx_bibblock">
Bengtsson, J., Eriksson, K. M., Hartmann, M., Wang, Z., Shenoy, B. D., Grelet,
G.-A., Abarenkov, K., Petri, A., Rosenblad, M. A., Nilsson, R. H. (2011)
Metaxa: a software tool for automated detection and discrimination among
ribosomal small subunit (12S/16S/18S) sequences of archaea, bacteria,
eukaryotes, mitochondria, and chloroplasts in metagenomes and environmental
sequencing datasets. <span class="ltx_text ltx_font_italic">Antonie Van Leeuwenhoek</span>, <span class="ltx_text ltx_font_bold">100</span>, 471–475.

</span>
      </li>
      
<li id="bib.bib4" class="ltx_bibitem" style="padding:0em 0em 0.2em 0em;">
        
        
        
        
        
<span class="ltx_bibblock">
Bengtsson-Palme, J., Ryberg, M., Hartmann, M., Branco, S., Wang, Z., Godhe, A.,
De Wit, P., Sánchez-García, M., Ebersberger, I., de Sousa, F., Amend,
A., Jumpponen, A., Unterseher, M., Kristiansson, E., Abarenkov, K., Bertrand,
Y. J. K., Sanli, K., Eriksson, K. M., Vik, U., Veldre, V., Nilsson, R. H.
(2013) Improved software detection and extraction of ITS1 and ITS2 from
ribosomal ITS sequences of fungi and other eukaryotes for analysis of
environmental sequencing data <span class="ltx_text ltx_font_italic">Methods in Ecology and Evolution</span>,
<span class="ltx_text ltx_font_bold">4</span>, 914–919.

</span>
      </li>
      
<li id="bib.bib5" class="ltx_bibitem" style="padding:0em 0em 0.2em 0em;">
        
        
        
        
        
<span class="ltx_bibblock">
Edgar, R. C., Haas, B. J., Clemente, J. C., Quince, C., Knight, R. (2011)
UCHIME improves sensitivity and speed of chimera detection. <span class="ltx_text ltx_font_italic">Bioinformatics</span>, <span class="ltx_text ltx_font_bold">27</span>, 2194–2200.

</span>
      </li>
      
<li id="bib.bib6" class="ltx_bibitem" style="padding:0em 0em 0.2em 0em;">
        
        
        
        
        
<span class="ltx_bibblock">
Fadrosh, D. W., Ma, B., Gajer, P., Sengamalay, N., Ott, S., Brotman, R. M.,
Ravel, J. (2014) An improved dual-indexing approach for multiplexed 16S rRNA
gene sequencing on the Illumina MiSeq platform. <span class="ltx_text ltx_font_italic">Microbiome</span>,
<span class="ltx_text ltx_font_bold">2</span>,  6.

</span>
      </li>
      
<li id="bib.bib7" class="ltx_bibitem" style="padding:0em 0em 0.2em 0em;">
        
        
        
        
        
<span class="ltx_bibblock">
Hamady, M., Walker, J. J., Harris, J. K., Gold, N. J., Knight, R. (2008)
Error-correcting barcoded primers for pyrosequencing hundreds of samples in
multiplex. <span class="ltx_text ltx_font_italic">Nature Methods</span>, <span class="ltx_text ltx_font_bold">5</span>, 235–237.

</span>
      </li>
      
<li id="bib.bib8" class="ltx_bibitem" style="padding:0em 0em 0.2em 0em;">
        
        
        
        
        
<span class="ltx_bibblock">
Huson, D. H., Auch, A. F., Qi, J., Schuster, S. C. (2007) MEGAN analysis of
metagenomic data. <span class="ltx_text ltx_font_italic">Genome Research</span>, <span class="ltx_text ltx_font_bold">17</span>, 377–386.

</span>
      </li>
      
<li id="bib.bib9" class="ltx_bibitem" style="padding:0em 0em 0.2em 0em;">
        
        
        
        
        
<span class="ltx_bibblock">
Illumina corporation (2013) 16S metagenomic sequencing library preparation.

</span>
      </li>
      
<li id="bib.bib10" class="ltx_bibitem" style="padding:0em 0em 0.2em 0em;">
        
        
        
        
        
<span class="ltx_bibblock">
Katoh, K., Standley, D. M. (2013) MAFFT multiple sequence alignment software
version 7: improvements in performance and usability. <span class="ltx_text ltx_font_italic">Molecular Biology
and Evolution</span>, <span class="ltx_text ltx_font_bold">30</span>, 772–780.

</span>
      </li>
      
<li id="bib.bib11" class="ltx_bibitem" style="padding:0em 0em 0.2em 0em;">
        
        
        
        
        
<span class="ltx_bibblock">
Kunin, V., Engelbrektson, A., Ochman, H., Hugenholtz, P. (2010) Wrinkles in the
rare biosphere: pyrosequencing errors can lead to artificial inflation of
diversity estimates. <span class="ltx_text ltx_font_italic">Environmental Microbiology</span>, <span class="ltx_text ltx_font_bold">12</span>,
118–123.

</span>
      </li>
      
<li id="bib.bib12" class="ltx_bibitem" style="padding:0em 0em 0.2em 0em;">
        
        
        
        
        
<span class="ltx_bibblock">
Lange, A., Jost, S., Heider, D., Bock, C., Budeus, B., Schilling, E.,
Strittmatter, A., Boenigk, J., Hoffmann, D. (2015) AmpliconDuo: A
Split-Sample Filtering Protocol for High-Throughput Amplicon Sequencing of
Microbial Communities. <span class="ltx_text ltx_font_italic">PLoS One</span>, <span class="ltx_text ltx_font_bold">10</span>, e0141590.

</span>
      </li>
      
<li id="bib.bib13" class="ltx_bibitem" style="padding:0em 0em 0.2em 0em;">
        
        
        
        
        
<span class="ltx_bibblock">
Larkin, M. A., Blackshields, G., Brown, N. P., Chenna, R., McGettigan, P. A.,
McWilliam, H., Valentin, F., Wallace, I. M., Wilm, A., Lopez, R., Thompson,
J. D., Gibson, T. J., Higgins, D. G. (2007) Clustal W and Clustal X version
2.0. <span class="ltx_text ltx_font_italic">Bioinformatics</span>, <span class="ltx_text ltx_font_bold">23</span>, 2947–2948.

</span>
      </li>
      
<li id="bib.bib14" class="ltx_bibitem" style="padding:0em 0em 0.2em 0em;">
        
        
        
        
        
<span class="ltx_bibblock">
Li, W., Fu, L., Niu, B., Wu, S., Wooley, J. (2012) Ultrafast clustering
algorithms for metagenomic sequence analysis. <span class="ltx_text ltx_font_italic">Briefings in
Bioinformatics</span>, <span class="ltx_text ltx_font_bold">13</span>, 656–668.

</span>
      </li>
      
<li id="bib.bib15" class="ltx_bibitem" style="padding:0em 0em 0.2em 0em;">
        
        
        
        
        
<span class="ltx_bibblock">
Nelson, M. C., Morrison, H. G., Benjamino, J., Grim, S. L., Graf, J. (2014)
Analysis, optimization and verification of Illumina-generated 16S rRNA gene
amplicon surveys. <span class="ltx_text ltx_font_italic">PLoS One</span>, <span class="ltx_text ltx_font_bold">9</span>, e94249.

</span>
      </li>
      
<li id="bib.bib16" class="ltx_bibitem" style="padding:0em 0em 0.2em 0em;">
        
        
        
        
        
<span class="ltx_bibblock">
Schmieder, R., Edwards, R. (2011) Quality control and preprocessing of
metagenomic datasets. <span class="ltx_text ltx_font_italic">Bioinformatics</span>, <span class="ltx_text ltx_font_bold">27</span>, 863–864.

</span>
      </li>
      
<li id="bib.bib17" class="ltx_bibitem" style="padding:0em 0em 0.2em 0em;">
        
        
        
        
        
<span class="ltx_bibblock">
Stevens, J. L., Jackson, R. L., Olson, J. B. (2013) Slowing PCR ramp speed
reduces chimera formation from environmental samples. <span class="ltx_text ltx_font_italic">Journal of
Microbiological Methods</span>, <span class="ltx_text ltx_font_bold">93</span>, 203–205.

</span>
      </li>
      
<li id="bib.bib18" class="ltx_bibitem" style="padding:0em 0em 0.2em 0em;">
        
        
        
        
        
<span class="ltx_bibblock">
Tanabe, A. S., Toju, H. (2013) Two new computational methods for universal DNA
barcoding: a benchmark using barcode sequences of bacteria, archaea, animals,
fungi, and land plants. <span class="ltx_text ltx_font_italic">PLoS One</span>, <span class="ltx_text ltx_font_bold">8</span>, e76910.

</span>
      </li>
      
<li id="bib.bib19" class="ltx_bibitem" style="padding:0em 0em 0.2em 0em;">
        
        
        
        
        
<span class="ltx_bibblock">
Zhang, J., Kobert, K., Flouri, T., Stamatakis, A. (2014) PEAR: a fast and
accurate Illumina Paired-End reAd mergeR. <span class="ltx_text ltx_font_italic">Bioinformatics</span>, <span class="ltx_text ltx_font_bold">30</span>,
614–620.

</span>
      </li>
    
</ul>
</section>
<section id="A1" class="ltx_appendix">
<h2 class="ltx_title ltx_title_appendix">
<span class="ltx_tag ltx_tag_appendix">Appendix A </span>Instllation of optional programs</h2>

<section id="A1.S1" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">A.1 </span>Installation of bcl2fastq</h3>

<div id="A1.S1.p1" class="ltx_para">
<p class="ltx_p">Illuminaから提供されている、BCL形式のベースコールデータからFASTQを生成するプログラムbcl2fastqはMiSeq・HiSeq用のv1.8.4およびNextSeq 500・HiSeq X用のv2.17.1.14が
<br class="ltx_break"><a href="http://support.illumina.com/downloads/bcl2fastq_conversion_software.html" title="" class="ltx_ref">http://support.illumina.com/downloads/bcl2fastq_conversion_software.html</a>
<br class="ltx_break">からダウンロードできます。
v1.8.4は以下のようにインストールできます。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># alienのインストール</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">$ sudo apt-get install alien</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># bcl2fastq v1.8.4のダウンロード</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">$ wget \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">ftp://webdata:webdata@ussd-ftp.illumina.com/Downloads/Software/bcl2fastq/bcl2fastq-1.8.4-Linux-x86_64.rpm</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># .rpmから.debへの変換とインストール</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">$ sudo alien -i \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">bcl2fastq-1.8.4-Linux-x86_64.rpm</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># bcl2fastqに必要な追加パッケージのインストール</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">$ sudo apt-get install libxml-simple-perl xsltproc</span>
          </span>
</p>
</div>
<div id="A1.S1.p2" class="ltx_para">
<p class="ltx_p">Ubuntu・Xubuntu・Lubuntuの14.04LTSでは、Perlのバージョンが新しすぎ、より古いバージョンを前提として書かれたbcl2fastqが動作しません。
そこで以下のようにして問題の部分を書き換えます。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">$ sudo perl -i -npe \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">'s/qw\(ELAND_FASTQ_FILES_PER_PROCESS\)/\("ELAND_FASTQ_FILES_PER_PROCESS"\)/' \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">/usr/local/lib/bcl2fastq-1.8.4/perl/Casava/Alignment/Config.pm</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">$ sudo perl -i -npe \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">'s/qw\(ELAND_GENOME\)/\("ELAND_GENOME"\)/' \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">/usr/local/lib/bcl2fastq-1.8.4/perl/Casava/Alignment/Config.pm</span>
          </span>
</p>
<p class="ltx_p">テキストエディタで
<br class="ltx_break"><span class="ltx_text ltx_font_typewriter">/usr/local/lib/bcl2fastq-1.8.4/perl/Casava/Alignment/Config.pm
<br class="ltx_break"></span>の747行目と751行目を手動で書き換えても構いません。
<span class="ltx_text ltx_font_typewriter">qw(FOO)</span>を<span class="ltx_text ltx_font_typewriter">("FOO")</span>に書き換えて下さい。</p>
</div>
<div id="A1.S1.p3" class="ltx_para">
<p class="ltx_p">v2.17.1.14のインストールをする場合は以下のようにコマンドを実行して下さい。</p>
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># rpm2cpioとcpioのインストール</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; sudo apt-get install rpm2cpio cpio↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># bcl2fastq2 v2.17のダウンロード</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; wget \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">ftp://webdata2:webdata2@ussd-ftp.illumina.com/downloads/software/bcl2fastq/bcl2fastq2-v2.17.1.14-Linux-x86_64.zip↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># 圧縮ファイルの展開</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; unzip -qq bcl2fastq2-v2.17.1.14-Linux-x86_64.zip↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># .rpmを展開してコマンドを抽出</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; rpm2cpio \</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">bcl2fastq2-v2.17.1.14-Linux-x86_64.rpm | cpio -id↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># コマンドをインストール</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; sudo mv usr/local/bin/bcl2fastq /usr/local/bin/↓</span>
          </span>
</p>
</div>
</section>
</section>
<section id="A2" class="ltx_appendix">
<h2 class="ltx_title ltx_title_appendix">
<span class="ltx_tag ltx_tag_appendix">Appendix B </span>Terminal command examples</h2>

<section id="A2.S1" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">B.1 </span>Counting sequences in a file</h3>

<div id="A2.S1.p1" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># FASTQの場合</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; grep -P -c '^\+$' inputfile↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># GZIP圧縮FASTQの場合</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; gzip -dc inputfile | grep -P -c '^\+$'↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># BZIP2圧縮FASTQの場合</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; bzip2 -dc inputfile | grep -P -c '^\+$'↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># XZ圧縮FASTQの場合</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; xz -dc inputfile | grep -P -c '^\+$'↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># FASTAの場合</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; grep -P -c '^&gt;' inputfile↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># GZIP圧縮FASTAの場合</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; gzip -dc inputfile | grep -P -c '^&gt;'↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># BZIP2圧縮FASTAの場合</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; bzip2 -dc inputfile | grep -P -c '^&gt;'↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># XZ圧縮FASTAの場合</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; xz -dc inputfile | grep -P -c '^&gt;'↓</span>
          </span>
</p>
</div>
</section>
<section id="A2.S2" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">B.2 </span>Viewing sequence files</h3>

<div id="A2.S2.p1" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># 無圧縮ファイル内容を画面に出力</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; cat inputfile↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># 無圧縮ファイル内容をスクロールしながら見る</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; less inputfile↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># GZIP圧縮ファイル内容を画面に出力</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; gzip -dc inputfile↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># GZIP圧縮ファイル内容をスクロールしながら見る</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; gzip -dc inputfile | less↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># FASTQの最初の1配列を画面に出力</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; head -n 4 inputfile↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># GZIP圧縮FASTQの最初の1配列を画面に出力</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; gzip -dc inputfile | head -n 4↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># FASTQの特定の配列を検索して画面に出力</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; grep -P -A 3 '^\@配列名の正規表現' inputfile↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># GZIP圧縮FASTQの特定の配列を検索して画面に出力</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; gzip -dc inputfile | grep -P -A 3 '^\@配列名の正規表現'↓</span>
          </span>
</p>
</div>
</section>
<section id="A2.S3" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">B.3 </span>Compression and decompression</h3>

<div id="A2.S3.p1" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># フォルダ内の全.fastq.gzを展開</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; for f in *.fastq.gz↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">do gzip -d $f↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">done↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># フォルダ内の全.fastq.gzを展開(サブフォルダ含む)</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; for f in `find . -name *.fastq.gz`↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">do gzip -d $f↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">done↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># 4つのCPUを使用してフォルダ内の全.fastq.gzを展開</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; ls *.fastq.gz | xargs -L 1 -P 4 gzip -d↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># 4つのCPUを使用してフォルダ内の全.fastq.gzを展開(サブフォルダ含む)</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; find . -name *.fastq.gz | xargs -L 1 -P 4 gzip -d↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># フォルダ内の全.fastqを圧縮</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; for f in *.fastq↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">do gzip $f↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">done↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># フォルダ内の全.fastqを圧縮(サブフォルダ含む)</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; for f in `find . -name *.fastq`↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">do gzip $f↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">done↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># 4つのCPUを使用してフォルダ内の全.fastqを圧縮</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; ls *.fastq | xargs -L 1 -P 4 gzip↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># 4つのCPUを使用してフォルダ内の全.fastqを圧縮(サブフォルダ含む)</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; find . -name *.fastq | xargs -L 1 -P 4 gzip↓</span>
          </span>
</p>
</div>
</section>
<section id="A2.S4" class="ltx_section">
<h3 class="ltx_title ltx_title_section">
<span class="ltx_tag ltx_tag_section">B.4 </span>Extraction and output of sequences</h3>

<div id="A2.S4.p1" class="ltx_para">
<p class="ltx_p ltx_align_left" style="border:1px solid #000000;padding:0.5em 0em 0.5em 0em;">
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># FASTQから最初の1万配列を抽出</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; head -n 40000 inputfile &gt; outputfile↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># FASTQから最後の1万配列を抽出</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; tail -n 40000 inputfile &gt; outputfile↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># GZIP圧縮FASTQから最初の1万配列を抽出してGZIP圧縮FASTQへ保存</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; gzip -dc inputfile | head -n 40000 | gzip -c &gt; outputfile↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># GZIP圧縮FASTQから最後の1万配列を抽出してGZIP圧縮FASTQへ保存</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; gzip -dc inputfile | tail -n 40000 | gzip -c &gt; outputfile↓</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;"># GZIP圧縮FASTQの特定の配列を検索してGZIP圧縮FASTQへ保存</span>
          </span>
<span class="ltx_p ltx_align_left">
            <span class="ltx_text ltx_font_typewriter" style="font-size:80%;">&gt; gzip -dc inputfile | grep -P -A 3 '^\@配列名の正規表現' | gzip -c &gt; outputfile↓</span>
          </span>
</p>
</div>
</section>
</section><div class="ltx_rdf" about="" property="dcterms:subject" content="Metagenome"></div>

</article>
</div>
</div>
</body>
</html>
